---
title: "DNA Metabarcoding Captures Temporal and Vertical Dynamics of Mesozooplankton
  Communities - Supplementary Data S3"
---

***Authors: Andreas Novotny (1,2), Caterina Rodrigues (1,2), Loïc Jacquemot (1,2), Rute B. G. Clemente-Carvalho (2), Rebecca S. Piercey (2), Evan Morien (2), Moira Galbraith (3), Colleen T. E. Kellogg (2), Matthew A. Lemay (2), and Brian P. V. Hunt (1,4)***

1.  *Institute for the Oceans and Fisheries, University of British Columbia, 2202 Main Mall, Vancouver, BC, V6T 1Z4 Canada*

2.  *Hakai Institute, PO Box 25039, Campbell River, BC, V9W 0B7, Canada*

3.  *Institute of Ocean Sciences, Fisheries and Oceans Canada, PO Box 6000, Sidney, BC, V8L 4B2, Canada*

4.  *Department of Earth, Ocean and Atmospheric Sciences, University of British Columbia, 2202 Main Mall, Vancouver, BC, V6T 1Z4 Canada*

## Abstract

In this study, we examined how well DNA metabarcoding capture changes in biomass of marine mesozooplankton. We evaluated the effectiveness of three different standardisation methods – relative abundance, presence-absence, and the eDNA-index – for correlating seasonal trends between DNA sequence read abundance and microscopic biomass estimates of zooplankton net samples collected every two weeks in a 1-year time series. To assess if taxa specific trends are sensitive to the selection of genetic marker, we sequenced both short fragment of the COI gene, with primers optimised to capture marine invertebrate community (Leray *et al.*, 2013) and the 18S gene, with universal primers optimised for marine microbial eukaryotes (Balzano *et al.*, 2015). In doing so, we provide both new insights into the reliability of DNA metabarcoding analysis and guidelines for future studies that aim to use environmental DNA to study marine community dynamics. Further, through analysis of discreet water column DNA samples, we demonstrate the potential of the DNA approach in resolving differences in vertical zooplankton distributions.

# 0. Preparations

**This R notebook was used to produce the graphical and statistical output** for the manuscript and its supplementary figures and tables. See manuscript for details on sampling, experimental design and sample processing.\
\
**Prior ro analysis in this script** 18S and COI amplicons have been sequenced on a MiSeq and raw Fastq sequences have been uploaded to NCBI under the BioProject accession number PRJNA1141475 (<https://www.ncbi.nlm.nih.gov/sra/PRJNA1141475>). All sample-specific accession numbers are listed in the metadata files. See further down.\
\
DNA sequences (FASTQ files) were processed using a custom bioinformatic pipeline (see Supplementary Data 2: <https://doi.org/10.5281/zenodo.14241795>) using DADA2, with sequences annotated to the MetaZooGene database. ASV tables with taxonomic annotations are found in the ./Data folder.

**These R packages are required for executing the code below:**

```{r message=FALSE}
# Data import
library(readxl)

# All data format modifications and filtering and plotting
library(tidyverse)
'%!in%' <- function(x,y)!('%in%'(x,y))

# This script contains a function for calculating eDNA-index.
source("Code/Functions.R")

# Specific plotting tools
library(UpSetR) # For making Euler diagrams
library(eulerr) # For making Euler diagrams
library(ggOceanMaps) # For maps
library(ggspatial)  # For map modifications
library(cowplot) # Simple plot themes
library(ggpubr) # Plot layout
library(patchwork) # Plot layout
library(ggnewscale) # Multiple scales on one plot

# Statistical analyses
library(vegan)

# Import all datasets from Bioinformatic Pipeline
#system2("cp", args = c(
#  "../CoastalBCdata/Data_output/To_ZoopMethodsComp/*",
#  "Data/"
#))
```

## Sampling Station

Samples were collected during 2017 from station QU39 in the Salish Sea, British Columbia, Canada.

```{r}
# 1. Draw inner map with points for stations

# Make a dataframe cointatining staion coordinates
pin <- data.frame(lon = c(-125.0992), 
                 lat = c(50.0307), 
                 label = c("QU39"))

# Plot inner map
inner <-
  # Define the limits (zoom of the map)
  basemap(limits = c(-126.5, -121.8, 48.3, 51.3), rotate = TRUE) +
  # Add red point for sampling station(s)
  geom_spatial_point(data = pin, aes(x = lon, y = lat), color = "red", size = 2) +
  # Add station labels, but adjuest position to not cover the point.
  geom_spatial_label(data = pin, aes(x = lon-0.35, y = lat-0.15, label = label)) +
  # Remove axis title
  theme(axis.title.x=element_blank(),
        axis.title.y=element_blank())


# 2. Draw outer map with box highlighting the inner map.

# Make a dataframe cointatining 4 coordinates defining box corners.
# Use same coordingates as the limits defined above.
insert_box <- data.frame(lon = c(-126.5, -126.5, -121.8, -121.8),
                         lat = c(48.3, 51.3, 51.3, 48.3))
# Plot outer map
outer <-
  # Define the limits (zoom of the map)
  basemap(limits = c(-130.0, -80.0, 35.0, 75.0), rotate = TRUE) +
  # Draw polygon for inner map
  geom_spatial_polygon(data = insert_box, aes(x = lon, y = lat),
                       fill = NA, color = "red") +
  # Remove axis title
  theme(axis.title.x=element_blank(),
        axis.title.y=element_blank())


# 3. Make outer plot as insert of inner. Define size and location.
final_map <- inner + inset_element(outer, 0.5, 0.5, 1, 1)
ggsave("Figure_output/Fig.1B_StationMap.pdf", final_map, width = 4, height = 4)

final_map

rm(inner, outer, insert_box, pin)
```

## Load all data

Reading in all COI ASVs with taxonomic annotation and sample metadata. 

```{r}
# Sample metadata
metadata_COI <- read_csv("Data/Sequence_Metadata_COI_QU39_2017.csv")
# ASV and taxonomy table
ASV_COI <- read_csv("Data/ASV_COI_MZGdb_QU39_2017.csv")

# Turn ASV table into "long" and merge with metadata.
All_COI_data <- ASV_COI %>%
  pivot_longer(cols = !1:13, names_to = "library_ID", values_to = "Abundance") %>% 
  full_join(metadata_COI)

rm(metadata_COI, ASV_COI)
```

Reading in all 18S ASVs with taxonomic annotation and sample metadata. 

```{r}
# Sample metadata
metadata_18S <- read_csv("Data/Sequence_Metadata_18S_QU39_2017.csv")
# ASV and taxonomy table
ASV_18S <- read_csv("Data/ASV_18S_MZGdb_QU39_2017.csv")

# Turn ASV table into "long" and merge with metadata.
All_18S_data <- ASV_18S %>%
  pivot_longer(cols = !1:13, names_to = "library_ID", values_to = "Abundance") %>% 
  full_join(metadata_18S)

rm(metadata_18S, ASV_18S)
```

For DNA, three different sample types were collected: Filtered Water, Bulk Zooplankton nets, and filtered ethanol preservative from the zooplankton nets. Here we construct a new Type variable that is based on the "collection_method" from the sample metadata.

```{r}
# 1. Get levels for collection method:
#All_COI_data %>% 
#  pull(collection_method) %>% unique()

#. 2. Rename the three levels with more useable names:
Filtered_Water <- "Sterivex Seawater filtration: https://dx.doi.org/10.17504/protocols.io.rm7vzjxrrlx1/v1"
Bulk_Zooplankton <- "Vertical zooplankton net tow, preserved in 99 percent ethanol"
Zooplankton_Preservative <- "Vertical zooplankton net tow, preserved in 99 percent ethanol. Etnahol subsampling: https://dx.doi.org/10.17504/protocols.io.j8nlk888wl5r/v1"

# Add new Type variable to COI metadata. 
All_COI_data <- All_COI_data %>% 
  mutate(Type = ifelse(
    collection_method == Filtered_Water,
    "Filtered_Water",
    ifelse(
      collection_method == Bulk_Zooplankton,
      "Bulk_Zooplankton",
      "Zooplankton_Preservative")
  ))

# Add new Type variable to 18S metadata.
All_18S_data <- All_18S_data %>% 
  mutate(Type = ifelse(
    collection_method == Filtered_Water,
    "Filtered_Water",
    ifelse(
      collection_method == Bulk_Zooplankton,
      "Bulk_Zooplankton",
      "Zooplankton_Preservative")
  ))
```

Reading in data from taxonomic zooplankton analysis.

```{r}
df_count_2017 <- read_csv("Data/Zooplankton_QU39_2017.csv")
```

# 1. Sample Quality Control

## Sequencing Depth
Libraries with poor sequencing depth are removed. Our threshold is set to 10,000 reads per sequencing library.
Further, in the MetaZooGene database, the genus Corycaeus was sometimes, but not always, labeled as Ditrichocorycaeus.
Ditrichocorycaeus is a potential subgenus, but is not yet commonly accepted. We correct that here.

For COI:

```{r}
# 1. Filter out libraries under threshold line of 10,000 reads per library
Good_COI_data <- All_COI_data %>% 
  group_by(library_ID) %>%
  filter(sum(Abundance)>10000) %>% 
  ungroup()

# 2.Summarize statistics of sequencing depth.  
print("Median COI library size:")
Good_COI_data %>%
  group_by(library_ID) %>% 
  summarise(SeqDepth = sum(Abundance)) %>%
  pull(SeqDepth) %>% median

print("Number of final COI libraries:")
Good_COI_data %>% 
  pull(library_ID) %>% unique %>% length


#### NOTE: Taxonomic correction of the subgenus Ditrichocorycaeus, not yet accepted.
Good_COI_data <- Good_COI_data %>%
  mutate(Genus = ifelse(Genus == "Ditrichocorycaeus", "Corycaeus", Genus),
         Species = ifelse(Species == "Ditrichocorycaeus anglicus","Corycaeus anglicus", Species))

```

Same for 18S:

```{r}

# Filter out libraries with low read counts.
Good_18S_data <- All_18S_data %>% 
  group_by(library_ID) %>% 
  filter(sum(Abundance)>10000) %>% 
  ungroup()


print("Median 18S library size:")
Good_18S_data %>% 
  group_by(library_ID) %>%
  summarise(SeqDepth = sum(Abundance)) %>%
  pull(SeqDepth) %>% median

print("Number of final 18S libraries:")
Good_18S_data %>% 
  pull(library_ID) %>% unique %>%  length

#### NOTE: Taxonomic correction of the subgenus Ditrichocorycaeus, not yet accepted.
Good_18S_data <- Good_18S_data %>%
  mutate(Genus = ifelse(Genus == "Ditrichocorycaeus", "Corycaeus", Genus),
         Species = ifelse(Species == "Ditrichocorycaeus anglicus","Corycaeus anglicus", Species))
```

Summarize tables of sample frequency and sequencing depth. Creating table for zooplankton samples:
```{r}
# 18S Zooplankton bulk data:
tmp_18S <- Good_18S_data %>%
  group_by(library_ID, source_material_id, collection_date, Type, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  filter(Type == "Bulk_Zooplankton") %>% 
  ungroup() %>% 
  select(collection_date, Abundance) %>% 
  mutate(Marker = "18S_Sequences")

# COI Zooplankton bulk data:
tmp_COI <- Good_COI_data %>%
  group_by(library_ID, source_material_id, collection_date, Type, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  filter(Type == "Bulk_Zooplankton") %>% 
  ungroup() %>% 
  select(collection_date, Abundance) %>% 
  mutate(Marker = "COI_Sequences")

# COI Zooplankton preservative
tmp_COI_EtOH <- Good_COI_data %>%
  group_by(library_ID, source_material_id, collection_date, Type, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  filter(Type == "Zooplankton_Preservative") %>% 
  ungroup() %>% 
  select(collection_date, Abundance) %>% 
  mutate(Marker = "COI_Sequences_(Sup)")

# Microscopy Zooplankton data
tmp_mic <- df_count_2017 %>%
  group_by(Sample_ID, Station, Date, STN_TIME) %>% 
  summarise(Biomass = sum(Biomass, na.rm = TRUE)) %>%
  ungroup() %>% 
  select(collection_date = Date, Abundance = Biomass) %>% 
  mutate(Marker = "Microscopy_Biomass")

# Combine and turn into table.
bind_rows(tmp_mic, tmp_18S, tmp_COI, tmp_COI_EtOH) %>% 
  pivot_wider(names_from = Marker, values_from = Abundance) %>% 
  arrange(collection_date) %>% 
  write_csv(file = "Data_output/Sup.Tab.S1_ZooplanktonSampleSummary.csv")
  
rm(tmp_mic, tmp_18S, tmp_COI, tmp_COI_EtOH)
```

Creating table for water samples:
```{r}
# 18S Water sample sequencing depth
tmp_18S <- Good_18S_data %>%
  group_by(library_ID, source_material_id, collection_date, Type, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  filter(Type == "Filtered_Water") %>% 
  ungroup() %>% 
  select(collection_date, Abundance, depth) %>% 
  mutate(Marker = "18S_Sequences")

# COI water sample sequencing depth
tmp_COI <- Good_COI_data %>%
  group_by(library_ID, source_material_id, collection_date, Type, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  filter(Type == "Filtered_Water") %>% 
  ungroup() %>% 
  select(collection_date, Abundance, depth) %>% 
  mutate(Marker = "COI_Sequences")

# Combine to table
bind_rows(tmp_18S, tmp_COI) %>%
  group_by(collection_date, depth, Marker) %>% 
  summarise(Abundance = sum(Abundance, na.rm = TRUE)) %>% 
  pivot_wider(names_from = Marker, values_from = Abundance) %>% 
  arrange(collection_date, as.numeric(depth)) %>%
  mutate(`Sequences COI/18S` = paste(COI_Sequences, `18S_Sequences`, sep = " / ")) %>% 
  select(collection_date, depth, `Sequences COI/18S`) %>% 
  pivot_wider(names_from = depth, values_from = `Sequences COI/18S`) %>% 
  write_csv2(file = "Data_output/Suppl.Tab.S2_WaterSampleSummary.csv")

rm(tmp_mic, tmp_18S, tmp_COI, tmp_COI_EtOH)
```

## Annotation success

This first plot shows the relative abundance of Metazooans, compared to other organisms for 18S and COI respectively. 

```{r}
# Calculate % metazoans for all samples in 18S data
summary_18S <- Good_18S_data %>%
  mutate(depth = ifelse(Type == "Bulk_Zooplankton", "230-0", depth)) %>%
  # Summarise abundance per kingdom
  group_by(collection_date, Kingdom, library_ID, depth) %>% 
  summarise(Abundance = sum(Abundance)) %>%
  # Calculate relative abuncance per sample
  group_by(collection_date, depth, library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Kingdom, Abundance) %>%
  mutate(Marker = "18S")


# Calculate % metazoans for all samples in COI data
summary_COI <- Good_COI_data %>% 
  mutate(depth = ifelse(Type == "Bulk_Zooplankton", "230-0", depth)) %>%
  # Summarise abundance per kingdom
  group_by(collection_date, depth, Kingdom, library_ID) %>% 
  summarise(Abundance = sum(Abundance)) %>%
  # Calculate relative abuncance per sample
  group_by(collection_date, depth, library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Kingdom, Abundance) %>%
  mutate(Marker = "COI")

# Merge the two datasets
bind_rows(summary_18S, summary_COI) %>%
  # Show only proportion of animals
  filter(Kingdom == "Animalia") %>%
  # Arrange sampling depths
  mutate(depth = ifelse(depth == "230", "260", depth)) %>% 
  mutate(depth = factor(depth,
                                   levels= c("0", "5", "30",
                                             "100", "260", "230-0"))) %>%
  # Plot boxplot
  ggplot() +
  geom_boxplot(aes(depth, RA, color = Marker)) +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  scale_color_manual(values = c('#377eb8','#4daf4a'))

ggsave("Figure_output/Sup.Fig.S1A_AnnotationSuccess.pdf", width = 4, height = 4)

rm(summary_18S, summary_COI)

```



# 2. Data Transformation

In this section the different data formats of the 2017 QU39 time series will be mutated into similar data shapes and transformed in such a way that the data types can be compared. To keep filtering parameters equal for all data types, a set of functions are first defined, that are then executed for all data sets. Comparisons are done both at species and at genus level.

Key for data transformation is the index_RA function that is defined in ./Code/Functions.R. This function calculates eDNA-index and relative abundance for all taxa and samples. Read more details in the function description.

## Transforming Microscopy data:

Microscopy count data from integrated zooplankton net tows:

```{r}
# Species level
Net_count_2017_index_species <- df_count_2017 %>% 
  mutate(Depth = DEPTH_STRT,
         Taxa = paste(Genus, Tax1, sep = " "),
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)

# Genus level
Net_count_2017_index_genus <- df_count_2017 %>% 
  mutate(Depth = DEPTH_STRT,
         Taxa = Genus,
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)
```

## Transforming DNA data:

The DNA datasets are transformed identical to the microscopy data sets. Due to their similarity, a generalized function is first created, then executed for the data sets and taxonomic levels individually.

```{r}
# Define wrapper function
calculate_index <- function(data, taxrank, type) {
  
  data %>% 
    # Filter Sample type
    filter(Type == type,
           # Only animals and annotated taxa are included
           Kingdom == "Animalia",
           str_detect({{ taxrank }}, "@", negate = TRUE)) %>%
    mutate(Taxa = {{ taxrank }},
           Depth = depth,
           Date = collection_date,
           Sample = library_ID) %>% 
    # Execute the transformation
    index_RA(Sample, Taxa, Abundance, Depth, Date)
}

# Execute function for all subsets:
Net_18S_2017_index_species <- calculate_index(Good_18S_data, Species, "Bulk_Zooplankton")
Net_18S_2017_index_genus <- calculate_index(Good_18S_data, Genus, "Bulk_Zooplankton")
eDNA_18S_2017_index_species <- calculate_index(Good_18S_data, Species, "Filtered_Water")
eDNA_18S_2017_index_genus <- calculate_index(Good_18S_data, Genus, "Filtered_Water")

Net_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "Bulk_Zooplankton")
Net_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "Bulk_Zooplankton")
eDNA_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "Filtered_Water")
eDNA_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "Filtered_Water")

EtOH_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "Zooplankton_Preservative")
EtOH_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "Zooplankton_Preservative")

```

Defining a function for filtering singletons. 

```{r}
ListNoSing <- function(data, type = "DNA") {
  
  if (type == "DNA") {data <- filter(data, Abundance > 1)}
  
  data %>%
  group_by(Taxa) %>% 
  summarise(n=n()) %>% 
  filter(n>1) %>% 
  pull(Taxa)
}
```


# 3. Taxonomic Comparison

We compared the range of taxa that were captured with the Marine-Invertebrate COI primer vs the universal Eukaryote 18S primer, compared with microscopic zooplankton analysis.

## 3.1 Euler plots

Comparing zooplankton net samples, 18S vs COI vs microscopy.

```{r}
# Genus level comp.
GenusListComp_1 <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_genus),
  `Net COI` = ListNoSing(Net_COI_2017_index_genus)
)

pdf("Figure_output/Fig.2A_EulerGenusNet.pdf", width = 3.5, height = 3.5)
  set.seed(102)
  GenusListComp_1 %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
         lty = 1:5)
  dev.off()
  
  

# Species level comp.
SpeciesListComp_1 <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_species, "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_species),
  `Net COI` = ListNoSing(Net_COI_2017_index_species)
)

pdf("Figure_output/Fig.2B_EulerSpeciesNet.pdf", width = 3.5, height = 3.5)
  SpeciesListComp_1 %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
        lty = 1:5)
  dev.off()

  

```

Comparing zooplankton net samples, with water samples. 

```{r}
# 18S and microscopy
ListComp_18S <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, type = "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_genus),
  `Water 18S` = ListNoSing(eDNA_18S_2017_index_genus)
)

pdf("Figure_output/Fig.S2A_EulerCOI.pdf", width = 3.5, height = 3.5)
  set.seed(102)
  ListComp_18S %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
         lty = 1:5)
  dev.off()

# COI and microscopy
ListComp_COI <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, type = "m"),
  `Net COI` = ListNoSing(Net_COI_2017_index_genus),
  `Water COI` = ListNoSing(eDNA_COI_2017_index_genus)
)


pdf("Figure_output/Fig.S2B_Euler18S.pdf", width = 3.5, height = 3.5)
  ListComp_COI %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
        lty = 1:5)
  dev.off()
  

  
rm(GenusListComp_1, SpeciesListComp_1, ListComp_18S, ListComp_COI)
```

### Conclusions:

-   At species level, COI appears to Identify conciderably more of the counted zooplankton taxa than 18S.

-   At genus level all methods align better. COI preforms better than 18S, and Net samples preform better than eDNA.

## 3.2 Proportion of detected biomass

From the Euler diagram it appears as only a fraction of counted zooplankton is detected by the DNA methods. However, the diagram values all taxa equally and does not account for the relative importance of the taxa. The following code extracts the biomass proportion of counted taxa that are also detected by COI and 18S respectively.




```{r}
{ ### Preparing the data at Species Level:
Sum_per_species <- Net_count_2017_index_species %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_species, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_species %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_species %>% 
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")

Detected_18S <- Sum_per_species %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_species %>% 
  filter(Taxa %in% c(ListNoSing(Net_18S_2017_index_species),
                     ListNoSing(Net_COI_2017_index_species))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

#Plot yearly average
fraction_species <- 
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Species level")
}


{ #Preparing the data at Genus Level:
Sum_per_genus <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_genus %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_genus %>% 
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")
  
Detected_18S <- Sum_per_genus %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_genus %>% 
  filter(Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

fraction_genus <-   
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Genus level")
}


# Combine and plot
Biomass_fraction_plot <- bind_rows(fraction_genus, fraction_species) %>%
  #mutate(Level = factor(Level, levels = c("Species level", "Genus level"))) %>% 
  ggplot(aes(Marker, Proportion)) +
  geom_violin(aes(color = Marker)) +
  cowplot::theme_minimal_hgrid(12) +
  facet_wrap("Level") +
  theme(aspect.ratio = 1,
        axis.title.x=element_blank())
Biomass_fraction_plot

ggsave("Figure_output/Fig.2CD_BiomassFraction.pdf", width = 7, height = 4.5)


rm(Biomass_fraction_plot, Detected_18S, Detected_COI, fraction_genus,
   fraction_species, Sum_per_day, Sum_per_day_all,
   Sum_per_species, Sum_per_genus)

```

### Conclusions:

1.  18S (Balzano) Ccannot be used to identify taxa at species level.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

## 3.3 Exclusive taxa

So the DNA methods combined targets 50-70% of zooplankton my biomass. Here we investigate the remaining genera, that remains un-targeted by the DNA methods. The Genera are ordered by relative biomass contribution.

```{r}
# Detected by BOTH microscopy and ny of the other methods
microscopy_and_DNA <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(., "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>% 
  
  group_by(Taxa) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  mutate(RA = Abundance/sum(Abundance)) %>% 
  filter(RA > 0.000) %>% 
  mutate(Type = "Microscopy and DNA")


# Only detected by microscopy
only_microscopy <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, type = "m"),
         Taxa %!in% ListNoSing(Net_18S_2017_index_genus),
         Taxa %!in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  
  group_by(Taxa, Date) %>% 
  summarize(RA = mean(RA)) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  filter(RA > 0.000) %>% 
  mutate(Type = "Only by Microscopy")


# Only detected by COI
only_COI <- Net_COI_2017_index_genus %>%
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  filter(Taxa %!in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by COI")
only_COI

# Only detected by 18S
only_18S <- Net_18S_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_genus),
         Taxa %!in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by 18S")

#Combine and plot
bind_rows(microscopy_and_DNA, only_microscopy, only_COI, only_18S) %>% 
  mutate(Type = factor(Type, levels = c(
    "Microscopy and DNA", "Only by Microscopy", "Only by COI", "Only by 18S"
  ))) %>%
  filter(RA > 0.001) %>%  ##### NOTE: REMOVED LOW COUNTS! #######
  ggplot() +
  geom_bar(aes(reorder(Taxa, -RA), RA), stat = "identity") +
  theme_minimal_hgrid(9) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,
                                   face = "italic")) +
  facet_wrap("Type", scales = "free_x") +
  ylab("Relative abundance") +
  xlab("Genus")

ggsave("Figure_output/Fig.3_RemainingTaxa.pdf")


rm(only_18S, only_COI, only_microscopy, microscopy_and_DNA)
```

## Conclusions

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

5.  Out of the biomass not detected by the molecular markers, larger taxa, such as amphipods, has the highest biomass.

6.  eDNA of COI and 18S catches many different metazooan taxa not targeted by microscopy. Most of these taxa are benthic mollusks, worms, echinoderms or cnidarians.

## Selection of taxa subsets. 

To allow for plotting a subset of taxa, we define 

```{r}
# Vector for all genera
GenusList <-
  Net_count_2017_index_genus %>%
  
  # Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  # Pull list
  pull(Taxa) %>% unique()


# Vector for the top 27 abundant genera
GenusList27 <-
  Net_count_2017_index_genus %>%
  
  # Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"), 
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  
  # Select the 27 most abundant taxa
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>% 
  arrange(-RA) %>% 
  mutate(count = 1:35) %>% 
  filter(count < 28) %>% 
  
  # Pull list
  pull(Taxa) %>% unique()

# Vector for the top 15 abundant genera
GenusList15 <-
  Net_count_2017_index_genus %>%
  
  # Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  
  # Select the 27 most abundant taxa
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>% 
  arrange(-RA) %>% 
  mutate(count = 1:35) %>% 
  filter(count < 16) %>%
  
  # Pull list
  pull(Taxa) %>% unique()


# At Species Level
SpeciesList <-
  Net_count_2017_index_species %>% 
  filter(Taxa %in% ListNoSing(., "m"),
         Taxa %in% c(ListNoSing(Net_COI_2017_index_species),
                     ListNoSing(eDNA_COI_2017_index_species))) %>% 
  pull(Taxa) %>% unique()

```

# 4. Performance of normalization methods

We compare the preformace of normalization methods (Relative abundance RA, Pressence/Absence PA, and eDNA-index RA-index) for the Zooplankton net samples, to compare microscopy, 18S and COI analysis methods. To do this, we first construct a combined dataset:

```{r}
Net_merged_data <-
  # Combine data sets
  bind_rows(
    mutate(Net_count_2017_index_genus, marker = "Microscopy"),
    mutate(Net_COI_2017_index_genus, marker = "COI"),
    mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  
  # Calculate pressence / Absence from the relative abundance
  mutate(PA = ifelse(RA>0.001, 1, 0)) %>% 
  
  # Remove unwanted taxa and columns
  filter(Taxa %in% GenusList) %>%
  select(Taxa, Date, Abundance, RA, RA_Index, PA, marker) %>% 
  
  # Wrangle the data set to make tidy for comparisons between Molecular/Microscopy
  # Pivot transformation method long format. 
  pivot_longer(3:6, names_to = "Index_Type", values_to = "value") %>% 
  
  # Pivot marker (18S, COI, microscopy) wide.
  pivot_wider(names_from = marker, values_from = value) %>%
  
  # Make pairwise comparison: 18S-microscopy vs COI-microscopy
  pivot_longer(5:6, names_to = "Gene_Marker", values_to = "Molecular") %>% 
  select(Taxa, Date, Gene_Marker, Index_Type, Microscopy, Molecular) %>%
  
  # Remove samples with no matches Molecular/Mic
  filter(is.na(Molecular) == FALSE,
         is.na(Microscopy) == FALSE)
```

## 4.1 Multivariate comparison

To evaluate the effect of analysis and normalization method on the merged datasets, a permutational multivariate analysis of variance (permanova) is conducted with 999 permutations based on the Bray-Curtis distances between samples in the community matrix using adonis from the vegan R package v. 2.6-4  (Oksanen et al., 2007). The results are visualized with 2-dimensional non-metric multidimensional scaling (NMDS) using the metaMDS function from the vegan package (allowing for 100 iterations). Mantel test was used to test for covariance between sample distance matrixes produced with microscopy vs molecular data using the mantel function in vegan with 9999 permutations.

### Multivariate Function

To be able to repeat multivariate statistics and plots, multivariate statistics are defined as a function: 

```{r}
NMDS <- function(data = Net_merged_data, Marker = "COI", Type = "RA") {
  
  # Construct community matrix dataframe 
  wide <- filter(data, Gene_Marker == Marker, Index_Type == Type) %>%
    select(Taxa, Date, Microscopy, Molecular) %>%
    pivot_longer(3:4, names_to = "Method", values_to = "Value") %>% 
    pivot_wider(names_from = Taxa, values_from = Value)
  
  ## MANTEL TEST
  # Matrix for molecular
  molmat <- wide %>% 
    filter(Method == "Molecular") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Matrix for microscopy 
  micmat <- wide %>% 
    filter(Method == "Microscopy") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Control that sample names are the same
  control <- ifelse(row.names(molmat) != row.names(micmat),"WARNING", "OK")
  ifelse("WARNING" %in% control, print("WARNING"), print("OK"))
  
  # Execute the mantel test
  m.mod <- mantel(vegdist(molmat), vegdist(micmat))
  
  ## PERMANOVA and NMDS
  # Combined community matrix
  mat <- wide %>% 
    mutate(Sample = paste(Date, Method, sep = "_")) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Execute permanova
  permanova <- adonis2(mat ~ Method+Date,
                       data = mutate(wide, Date = format(as.Date(Date),
                                                         format="%m")),
                       permutations = 999, method="bray")
  
  # Execute NMDS
  mod <- metaMDS(mat, trymax = 100)
  
  
  # DATA EXTRACTION
  
  # Extract score coordinates from NMDS
  NMDS_scores <- function(mod) {
    Samples <- 
    scores(mod)$sites %>% 
    as.data.frame() %>%
    rownames_to_column("Name") %>% 
    separate(Name, into = c("Date", "Method"), sep = "_") %>% 
    mutate(Layer = "Samples")

  Taxa <-
    scores(mod)$species %>% 
    as.data.frame() %>%
    rownames_to_column("Taxa") %>% 
    mutate(Layer = "Taxa")

  bind_rows(Samples, Taxa)
  } # End NMDS Scores
  
  out <- NMDS_scores(mod) %>% 
    mutate(Gene_Marker = Marker, Index_Type = Type)
  
  
  # Combine data set of statistical output
  stats <- tibble(Marker = Marker,
       Type = Type,
       NMDS.stress = mod$stress,
       Method.R2 = permanova$R2[1],
       Method.F = permanova$F[1],
       Method.P = permanova$`Pr(>F)`[1],
       Date.R2 = permanova$R2[2],
       Date.F = permanova$F[2],
       Date.P = permanova$`Pr(>F)`[2],
       Mantel.R = m.mod$statistic,
       Mantel.P = m.mod$signif)
  
  output <- list(NMDS.scores = out,
                 NMDS.stressplot = stressplot(mod),
                 Stats = stats)
  
  
  return(output)
}

```

### Execute for all datasets

```{r}
mod1 <- NMDS(Marker = "COI", Type = "RA")
mod2 <- NMDS(Marker = "COI", Type = "PA")
mod3 <- NMDS(Marker = "COI", Type = "RA_Index")
mod4 <- NMDS(Marker = "18S", Type = "RA")
mod5 <- NMDS(Marker = "18S", Type = "PA")
mod6 <- NMDS(Marker = "18S", Type = "RA_Index")


bind_rows(
  mod1$Stats, mod2$Stats, mod3$Stats, mod4$Stats, mod5$Stats, mod6$Stats) %>% 
  write_csv(file = "./Figure_output/Tab.1_MultivariateStats.csv")

```

### Plot NMDS

We use the combined output from abuve to plot NMDS:

```{r}
# Define a seasonal color scheme:
seasonal_colors = c("#053061", "#3288BD", "#66C2A5", "#7FBC41",
                    "#A6D96A", "#FEE08B", "#FDAE61", "#F46D43",
                    "#D53E4F", "#9E0142", "#67001F", "#40004B")


# Combine multivariate outputs:
bind_rows(mod1$NMDS.scores, mod2$NMDS.scores, mod3$NMDS.scores,
          mod4$NMDS.scores, mod5$NMDS.scores, mod6$NMDS.scores) %>% 
  pivot_wider(names_from = Layer,
              values_from = c(NMDS1, NMDS2)) %>%
  # Modify scales and factors
  mutate(Month = format(as.Date(Date), format="%m")) %>%
  mutate(Index_Type = ifelse(Index_Type == "RA_Index", "eDNA Index", Index_Type)) %>% 
  mutate(Index_Type = factor(Index_Type, levels = c("RA", "PA", "eDNA Index"))) %>% 
  mutate(Taxa = ifelse(Taxa %in% GenusList15, Taxa, "")) %>% 
  
  # Plot
  ggplot(aes(NMDS1_Samples, NMDS2_Samples)) +
  
  # Taxa are represented as text
  geom_text(aes(NMDS1_Taxa, NMDS2_Taxa, label = Taxa),
            size = 3, colour = "grey", fontface = "italic") +
  
  # Dotted lines combine microscopy and DNA samples from the same date
  geom_line(aes(color = Month, group = Date), linetype = "dotted") +
  
  # Points represent each net.
  geom_point(aes(shape = Method, color = Month), size = 2) +
  
  # Eclipses for CI of the method (DNA vs Microscopy)
  stat_ellipse(aes(linetype = Method), size = 0.2) +
  
  # Facet Rows and Columns
  facet_grid(Index_Type~Gene_Marker, scales = "free") +
  
  # Esthetics
  theme_minimal_grid(12) +
  theme(aspect.ratio = 1,
        panel.border = element_rect(colour = "black", fill=NA),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  scale_color_manual(values = seasonal_colors) +
  scale_shape_manual(values=c(1, 16))

ggsave("Figure_output/Fig.3_NMDS.pdf", width = 8, height = 10)


```

## 4.2 Regression analyses

linear regressions between microscopy and molecular data were modelled, using the lm function.

### Plot correlations


Both 18S and COI combined, Relative abundance vs eDNA-index

```{r}
Net_merged_data %>% 
  filter(Index_Type %in% c("RA_Index", "RA")) %>%
  #filter(Molecular > 0, Microscopy > 0) %>% 
  ggplot(aes(Microscopy, Molecular)) +
  geom_point() +
  geom_smooth(aes(group = Taxa), color = "grey", method = lm, se=F) +
  geom_smooth(color = "#984ea3", method = lm, se=F) +
  theme_minimal_grid() +
  facet_wrap("Index_Type", scales = "free") +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey')) +
  ylim(0,1) +
  xlim(0,1)


ggsave("Figure_output/Fig.S3AB_Regression.pdf", width = 7, height = 4)

```

Calculate Rsq, P and Slope for all regressions:

```{r}
#Create new tables for regression summary and calculate coefficients 
# Empty table
species_list <- data.frame(Taxa = unique(Net_merged_data$Taxa),
                           RA_Rsq = 0,
                           #I_Rsq = 0,
                           
                           RA_p.val = 0,
                           #I_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           
                           RA_slope = 0,
                           I_slope = 0)

# Fill the table for each pairwise comparison:
for(i in 1:length(species_list$Taxa)){ #Start of loop
  RA <- Net_merged_data %>%
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA")
  RA_I <- Net_merged_data %>% 
    filter(Molecular > 0, Microscopy > 0) %>%
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA_Index")
  
  # Calculate the metrics:
  lm1 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA)
  lm2 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA_I)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  #species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  #species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]

}

species_list %>% 
  write_csv("./Figure_output/Tab.2_Regressions.csv")
```

These taxa have significant correlations:
```{r}
species_list %>%
  filter(RA_p.val > 0.05) %>% 
  pull(Taxa)
```

Whuile correlation coefficients are the same using RA and RA-index, the slope will differ. Here we plot the difference in slope between the two standardisation methods.
```{r}
species_list %>%
  filter(RA_p.val<0.05) %>% 
  select(RA_slope, I_slope) %>%
  pivot_longer(1:2) %>%
  mutate(Method = factor(name, levels = c("RA_slope", "I_slope")),
         `Slope (Molecular ~ Microscopy)` = value) %>%
  ggplot(aes(Method, `Slope (Molecular ~ Microscopy)`)) +
  geom_violin() +
  geom_point() +
  theme_minimal_hgrid() +
  scale_y_log10() +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey'))

ggsave("./Figure_output/Fig.S3C_SlopeDifferences.pdf", width = 4, height = 5)
```

# 5. Temporal comparison of taxonomic markers

The aim of this section is to compare if seasonal patterns of RA_index overlap between the different markers and methods. Zooplankton net tows (Count, COI and 18S) that are depth integrated (only one sample per sampling date) are plotted differently from the eDNA samples (that has a depth resolution with several samples per sampling date).\
Genus level comparison:

## Zooplankton Seasonality

Plot seasonal trend of bulk zooplankton samples

```{r}

# Combine the zooplankton datasets at genus level
tmp <- bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) 

# Arrange the taxa according to there total annual biomass
order <- tmp %>% 
  group_by(Taxa, marker) %>% 
  summarize(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>% 
  arrange(-Count) %>% 
  pull(Taxa)

tmp %>% 
  # Filter taxa (all) and order factors
  filter(Taxa %in% GenusList) %>%
  mutate(marker = factor(marker, levels = c("Count", "COI", "18S"))) %>%
  mutate(Taxa = factor(Taxa, levels = c(order))) %>% 
  
  # Plot
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker, linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 4) +
  
  # Esthetics
  theme_minimal_hgrid() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5) +
  scale_color_manual(values = c('#fec44f','#4daf4a','#377eb8'))

ggsave("Figure_output/Fig.5_ZooplanktonSeasonality.pdf", width = 8, height = 10)
```

RA_index of eDNA samples, the average of 18S and COI.

## Zooplankton Vertical Distribution

Plot verical distribution of zooplankton over the season.

```{r}
# Combine and merge COI and 18S eDNA components. Calculate mean of COI and 18S
eDNA <- bind_rows(mutate(eDNA_COI_2017_index_genus, marker = "COI"),
                  mutate(eDNA_18S_2017_index_genus, marker = "18S")) %>% 
  group_by(Taxa, Date, Depth) %>% 
  summarize(RA_Index = mean(RA_Index),
            Abundance = mean(Abundance),
            marker = "eDNA")

# Combine eDNA data with Net data components.
data <-  eDNA %>%
  bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
             mutate(Net_COI_2017_index_genus, marker = "COI"),
             mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  filter(Taxa %in% GenusList,
         Taxa %in% pull(eDNA, Taxa)) %>% 
  mutate(Depth = as.character(Depth)) %>% 
  
  # Modify the parameter shown on the categorical Y axis. That is depth for eDNA samples and Marker for Net samples". 
  mutate(y = ifelse(marker == "eDNA", Depth, marker)) %>% 
  mutate(y = factor(y, levels = rev(c("Count", "COI", "18S", "0", "5", "30", "100", "260")))) %>% 
  mutate(Taxa = factor(Taxa, levels = order))

data %>%  ggplot() +
  
  # Add filter water data:
  geom_point(
    mapping = aes(Date, y, color = RA_Index),
    #data = filter(data, y %!in% c("Count", "COI", "18S")),
    shape = 15) +
  scale_color_gradient2(
    high = '#984ea3', low = "white") +

  # Add Microscopy data
  new_scale_color() +
  geom_point(
    mapping = aes(Date, y, color = RA_Index),
    data = filter(data, y %in% c("Count")),
    shape = 15) +
  scale_color_gradient2(high = '#fec44f',
                       low = "white") +

  # Add Net COI data:
  new_scale_color() +
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("COI")),
    shape = 15) +
  scale_color_gradient2(high = '#4daf4a',
                      low = "white") +
  
  # Add Net 18S data:
  new_scale_color() +
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("18S")),
    shape = 15) +
  scale_color_gradient2(high = '#377eb8',
                       low = "white") +
  
  
  # Set plot themes
  #theme_map() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
  #      legend.position="none",
  #      panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.rtio = 1/2.5) +
  facet_wrap("Taxa", ncol = 3)

ggsave("Figure_output/Fig.6_VertivalSeasonCom.pdf", width = 7, height = 10)
```

## Species level comparison


The robustness of the eDNA index transformation can be visualized by repeating the trends at species level, and also including other sample techniques.

```{r}

# Combining datasets

tmp <- bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
          mutate(Net_COI_2017_index_species, marker = "COI_bulk"),
          mutate(EtOH_COI_2017_index_species, marker = "COI_sup"))

# Order by biomass
order <- tmp %>% 
  group_by(Taxa, marker) %>% 
  summarize(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>% 
  arrange(-Count) %>% 
  pull(Taxa)

# Arrange factors and plot. 
tmp %>%
  filter(Taxa %in% SpeciesList) %>%
  mutate(marker = factor(marker, levels = c("Count", "COI_bulk", "COI_sup"))) %>%
  mutate(Taxa = factor(Taxa, levels = order)) %>% 
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker,  linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 4) +
  
  # Esthetics
  theme_minimal_hgrid(8) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        #legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5,
        legend.position = "none") +
  scale_color_manual(values = c('#fec44f','#4daf4a',"cyan"))

ggsave("Figure_output/Supp.Fig.S5_SeasonalBulkSpecies.pdf", width = 10, height = 7)

```

## Data ammount

The ammount of sequences per taxa can hint at the reproducibility of the seasonal trends of the zooplankton. Species with low data may have less reliable results.

```{r}
  # Combine data sets
  bind_rows(
    mutate(Net_count_2017_index_genus, marker = "Microscopy"),
    mutate(Net_COI_2017_index_genus, marker = "COI"),
    mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 

  filter(Taxa %in% GenusList) %>%
  select(Taxa, Date, Abundance, marker) %>%
  group_by(Taxa, marker) %>%
  summarise(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>%
  arrange(-Microscopy) %>% 
  write_csv("Figure_output/Suppl.Tab.S4_DataAmmount.csv")
```
# 6. Session info

These packages and versions were used in this script. 

```{r}
sessionInfo()
```
