---
title: "Comparing Methods"
---

# 0. Background

**Andreas Novotny 2024**

This script is used to produce all graphical and statistical output for the manuscript and its supplementary material. Before starting this script, run all steps in the "prepare_datasets.Rmd"


These packages are required for this script.

```{r}
# Data import
library(readxl)

# All data format modifications and filtering and plotting
library(tidyverse)
'%!in%' <- function(x,y)!('%in%'(x,y))
source("Code/Functions.R")

# Specific plotting tools
library(UpSetR) # For making Euler diagrams
library(eulerr) # For making Euler diagrams
library(ggOceanMaps) # For maps
library(ggspatial)  # For map modifications
library(cowplot) # Simple plot themes
library(ggpubr) # Plot layout
library(patchwork) # Plot layout
library(ggnewscale) # Multiple scales on one plot

# Statistical analyses
library(vegan)

```



```{r}
# 1. Draw inner map with points for stations

# Make a dataframe cointatining staion coordinates
pin <- data.frame(lon = c(-125.0992), 
                 lat = c(50.0307), 
                 label = c("QU39"))

# Plot inner map
inner <-
  # Define the limits (zoom of the map)
  basemap(limits = c(-126.5, -121.8, 48.3, 51.3), rotate = TRUE) +
  # Add red point for sampling station(s)
  geom_spatial_point(data = pin, aes(x = lon, y = lat), color = "red", size = 2) +
  # Add station labels, but adjuest position to not cover the point.
  geom_spatial_label(data = pin, aes(x = lon-0.35, y = lat-0.15, label = label)) +
  # Remove axis title
  theme(axis.title.x=element_blank(),
        axis.title.y=element_blank())


# 2. Draw outer map with box highlighting the inner map.

# Make a dataframe cointatining 4 coordinates defining box corners.
# Use same coordingates as the limits defined above.
insert_box <- data.frame(lon = c(-126.5, -126.5, -121.8, -121.8),
                         lat = c(48.3, 51.3, 51.3, 48.3))
# Plot outer map
outer <-
  # Define the limits (zoom of the map)
  basemap(limits = c(-130.0, -80.0, 35.0, 75.0), rotate = TRUE) +
  # Draw polygon for inner map
  geom_spatial_polygon(data = insert_box, aes(x = lon, y = lat),
                       fill = NA, color = "red") +
  # Remove axis title
  theme(axis.title.x=element_blank(),
        axis.title.y=element_blank())


# 3. Make outer plot as insert of inner. Define size and location.
final_map <- inner + inset_element(outer, 0.5, 0.5, 1, 1)
ggsave("Figure_output/station_map.pdf", final_map, width = 4, height = 4)

final_map
```


# 1. Sequencing dept and negative controls.

In this section, we have a look at sequenced negative controls for the 18S and the COI libraries, and remove samples that have sequence depth lower than the negative controls.\
Here we also produce statistics for annotation levels and read success.\
\
Plotting sequencing depth and removing low libraries for COI:

```{r}
# Plot ranked sequencing depth with threshold line
All_COI_data %>% 
  seq_depth_plot() +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  geom_abline(intercept = 10000, linetype = 4)
ggsave("Figure_output/Sequencing_Depth_COI.pdf", width = 7, height = 4.5)


# Filter out libraries under threshold line
Good_COI_data <- All_COI_data %>% 
  group_by(Library_ID) %>%
  filter(Site_ID == "QU39") %>% 
  filter(sum(Abundance)>10000) %>% 
  ungroup()

print("Median COI library size:")
Good_COI_data %>% 
  group_by(Library_ID) %>% 
  summarise(SeqDepth = sum(Abundance)) %>%
  pull(SeqDepth) %>% median

print("Number of final COI libraries:")
Good_COI_data %>% 
  pull(Library_ID) %>% unique %>%  length
```

Plotting sequencing depth and removing low libraries for 18S:

```{r}
# Plot ranked sequencing depth
All_18S_data %>% 
  seq_depth_plot() +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  geom_abline(intercept = 10000, linetype = 4)
ggsave("Figure_output/Sequencing_Depth_18S.pdf" , width = 7, height = 4.5)

# Filter out libraries with low read counts.
Good_18S_data <- All_18S_data %>% 
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>10000) %>% 
  ungroup() 

print("Median 18S library size:")
Good_18S_data %>% 
  group_by(Library_ID) %>%
  filter(Site_ID == "QU39") %>% 
  summarise(SeqDepth = sum(Abundance)) %>%
  pull(SeqDepth) %>%median

print("Number of final 18S libraries:")
Good_18S_data %>% 
  pull(Library_ID) %>% unique %>%  length
```





## Annotation success

This first plot shows the relative abundance of Metazooans, compared to other organisms for 18S and COI.

```{r}
# Calculate % metazoans for all samples in 18S data
summary_18S <- Good_18S_data %>% 
  filter(Site_ID == "QU39",
         year(Sample_date) == 2017) %>% 
  mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>% 
  mutate(Sampling_depth = ifelse(Project_name == "Vertical", "230-0", Line_out_depth)) %>% 
  group_by(Sample_date, Sampling_depth, Kingdom, Library_ID) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  group_by(Sample_date, Sampling_depth, Library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Kingdom, Abundance) %>%
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>0) %>% 
  mutate(Marker = "18S")

# Calculate % metazoans for all samples in COI data
summary_COI <- Good_COI_data %>% 
  filter(Site_ID == "QU39",
         year(Sample_date) == 2017) %>% 
  mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>% 
  mutate(Sampling_depth = ifelse(Project_name == "Vertical", "230-0", Line_out_depth)) %>%
  filter(is.na(Sampling_depth)==FALSE) %>% 
  group_by(Sample_date, Sampling_depth, Kingdom, Library_ID) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  group_by(Sample_date, Sampling_depth, Library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Kingdom, Abundance) %>%
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>0) %>% 
  mutate(Marker = "COI")

# Merge and plot
bind_rows(summary_18S, summary_COI) %>%
  filter(Kingdom == "Animalia") %>% 
  mutate(Sampling_depth = ifelse(Sampling_depth == "230", "260", Sampling_depth)) %>% 
  mutate(Sampling_depth = factor(Sampling_depth,
                                   levels= c("0", "5", "30",
                                             "100", "260", "230-0"))) %>%
  ggplot() +
  geom_boxplot(aes(Sampling_depth, RA, color = Marker)) +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  scale_color_manual(values = c('#377eb8','#4daf4a'))

ggsave("Figure_output/Metazoan_reads.pdf", width = 4, height = 4)

```

This plot shows the distribution of annotation success for COI and 18S data, for all taxonomic levels in the MetaZoogene database. 
```{r}
success_18S <- Good_18S_data %>% 
  mutate(Annotation_level = as.numeric(Annotation_level)) %>% 
  mutate(ASV = paste(MiSeq_library, ASV, sep = "_")) %>% 
  group_by(ASV) %>% 
  summarise(Annotation_level = mean(Annotation_level)) %>% 
  group_by(Annotation_level) %>% 
  summarise(n = n()) %>% 
  mutate(marker = "18S") %>% 
  mutate(n = n/sum(n))

success_COI <- Good_COI_data %>% 
  mutate(Annotation_level = as.numeric(Annotation_level)) %>% 
  mutate(ASV = paste(MiSeq_library, ASV, sep = "_")) %>% 
  group_by(ASV) %>% 
  summarise(Annotation_level = mean(Annotation_level)) %>% 
  group_by(Annotation_level) %>% 
  summarise(n = n()) %>% 
  mutate(marker = "COI") %>% 
  mutate(n = n/sum(n))
success_COI

bind_rows(success_18S, success_COI) %>% 
  ggplot() +
  geom_bar(aes(Annotation_level, n, fill = marker),
           stat = "identity", position = "dodge") +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  scale_fill_manual(values = c('#377eb8','#4daf4a'))
ggsave("Figure_output/Annotation_Depth.pdf", width = 4, height = 4)

```

***RESULT***: COI has a notably higher percentage of Metazooan reads annotated, compared to 18S. Hoverver, this distinction is only present for water filers and not for barcoded zooplankton net samples.

# 2. Subset and calculate eDNA indexes of data sets:

In this section the different data formats of the 2017 QU39 time series will be mutated into similar data shapes and transformed in such a way that the data types can be compared. To keep filtering parameters equal for all data types, a set of functions are first defined, that are then executed for all data sets. Comparisons are done both at species and at genus level.

### Transforming Microscopy data:

Microscopy count data from integrated zooplankton net tows:

```{r}
# Species
Net_count_2017_index_species <- df_count_2017 %>% 
  filter(Station %in% c("QU39"),
         year(Date) == 2017) %>%
  mutate(Depth = DEPTH_STRT,
         Taxa = paste(Genus, Tax1, sep = " "),
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)

# Genus
Net_count_2017_index_genus <- df_count_2017 %>% 
  filter(Station %in% c("QU39"),
         year(Date) == 2017) %>%
  mutate(Depth = DEPTH_STRT,
         Taxa = Genus,
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)
```

### Transforming DNA data:

The DNA datasets are transformed identical to the microscopy data sets. Due to their similarity, a generalized function is first created, then executed for the data sets and taxonomic levels individually.

```{r}
calculate_index <- function(data, taxrank, project) {
  
  data %>% 
    filter(Project_name == project,
           year(Sample_date) == 2017,
           Kingdom == "Animalia",
           str_detect({{ taxrank }}, "@", negate = TRUE)) %>%
    mutate(Taxa = {{ taxrank }},
           Depth = Line_out_depth,
           Date = Sample_date,
           Sample = Library_ID) %>% 
    index_RA(Sample, Taxa, Abundance, Depth, Date)
}

Net_18S_2017_index_species <- calculate_index(Good_18S_data, Species, "Vertical")
Net_18S_2017_index_genus <- calculate_index(Good_18S_data, Genus, "Vertical")
eDNA_18S_2017_index_species <- calculate_index(Good_18S_data, Species, "QU39")
eDNA_18S_2017_index_genus <- calculate_index(Good_18S_data, Genus, "QU39")

Net_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "Vertical")
Net_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "Vertical")
eDNA_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "QU39")
eDNA_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "QU39")

EtOH_COI_2017_index_species <- calculate_index(Good_COI_data, Species, "EtOH_sup")
EtOH_COI_2017_index_genus <- calculate_index(Good_COI_data, Genus, "EtOH_sup")

```
Filter singeltons function

```{r}
ListNoSing <- function(data, type = "DNA") {
  
  if (type == "DNA") {data <- filter(data, Abundance > 1)}
  
  data %>%
  group_by(Taxa) %>% 
  summarise(n=n()) %>% 
  filter(n>1) %>% 
  pull(Taxa)
  
}


```



# 3. Coverage comparison of taxonomic markers


### Eulerr plots:

18S vs COI

```{r}
GenusListComp_1 <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_genus),
  `Net COI` = ListNoSing(Net_COI_2017_index_genus)
)

SpeciesListComp_1 <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_species, "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_species),
  `Net COI` = ListNoSing(Net_COI_2017_index_species)
)

  
pdf("Figure_output/euler_species.pdf", width = 3.5, height = 3.5)
  SpeciesListComp_1 %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
        lty = 1:5)
  dev.off()
  
pdf("Figure_output/euler_genus.pdf", width = 3.5, height = 3.5)
  set.seed(102)
  GenusListComp_1 %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
         lty = 1:5)
  dev.off()
```

Net vs water

```{r}
ListComp_18S <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, type = "m"),
  `Net 18S` = ListNoSing(Net_18S_2017_index_genus),
  `Water 18S` = ListNoSing(eDNA_18S_2017_index_genus)
)

ListComp_COI <- list(
  `Net Microscopy` = ListNoSing(Net_count_2017_index_genus, type = "m"),
  `Net COI` = ListNoSing(Net_COI_2017_index_genus),
  `Water COI` = ListNoSing(eDNA_COI_2017_index_genus)
)

  
pdf("Figure_output/euler_COI.pdf", width = 3.5, height = 3.5)
  ListComp_COI %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
        lty = 1:5)
  dev.off()
  
pdf("Figure_output/euler_18S.pdf", width = 3.5, height = 3.5)
  set.seed(102)
  ListComp_18S %>% 
    fromList() %>% 
    euler(shape = "circle") %>% 
    plot(quantities = TRUE,
         lty = 1:5)
  dev.off()
```


### Conclusions:

-   At species level, COI appears to Identify conciderably more of the counted zooplankton taxa than 18S.

-   At genus level all methods align better. COI preforms better than 18S, and Net samples preform better than eDNA.

## 3.3 Proportion of counted biomass detected by eDNA

From the Euler diagram it appears as only a fraction of counted zooplankton is detected by the DNA methods. However, the diagram values all taxa equally and does not account for the relative importance of the taxa. The following code extracts the biomass proportion of counted taxa that are also detected by COI and 18S respectively.

### Prepare data

At Species Level:

```{r}
Sum_per_species <- Net_count_2017_index_species %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_species, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_species %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_species %>% 
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")

Detected_18S <- Sum_per_species %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_species %>% 
  filter(Taxa %in% c(ListNoSing(Net_18S_2017_index_species),
                     ListNoSing(Net_COI_2017_index_species))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

#Plot yearly average
fraction_species <- 
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Species level")
```

At Genus Level:

```{r}
{ ##Species level biomass comparison##
Sum_per_genus <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_genus %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_genus %>% 
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")
  
Detected_18S <- Sum_per_genus %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_genus %>% 
  filter(Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

fraction_genus <-   
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Genus level")
}


```

### Plot

```{r}
# Combine and plot
Biomass_fraction_plot <- bind_rows(fraction_genus, fraction_species) %>%
  #mutate(Level = factor(Level, levels = c("Species level", "Genus level"))) %>% 
  ggplot(aes(Marker, Proportion)) +
  geom_violin(aes(color = Marker)) +
  cowplot::theme_minimal_hgrid(12) +
  facet_wrap("Level") +
  theme(aspect.ratio = 1,
        axis.title.x=element_blank())
Biomass_fraction_plot

ggsave("Figure_output/Biomass_fractions.pdf", width = 7, height = 4.5)


# Mean Results
bind_rows(fraction_genus, fraction_species) %>% 
  group_by(Marker, Level) %>% 
  summarize(Proportion = mean(Proportion))

```

### Conclusions:

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.


## 3.4. Exclusive taxa

So the DNA methods combined targets 50-70% of zooplankton my biomass. Here we investigate the remaining genera, that remains un-targeted by the DNA methods. The Genera are ordered by relative biomass contribution.

```{r}
# Detected by BOTH microscopy and ny of the other methods
microscopy_and_DNA <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(., "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>% 
  
  group_by(Taxa) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  mutate(RA = Abundance/sum(Abundance)) %>% 
  filter(RA > 0.000) %>% 
  mutate(Type = "Microscopy and DNA")


# Only detected by microscopy
only_microscopy <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, type = "m"),
         Taxa %!in% ListNoSing(Net_18S_2017_index_genus),
         Taxa %!in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  
  group_by(Taxa, Date) %>% 
  summarize(RA = mean(RA)) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  filter(RA > 0.000) %>% 
  mutate(Type = "Only by Microscopy")


# Only detected by COI
only_COI <- Net_COI_2017_index_genus %>%
  filter(Taxa %in% ListNoSing(Net_COI_2017_index_genus)) %>% 
  filter(Taxa %!in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by COI")
only_COI

# Only detected by 18S
only_18S <- Net_18S_2017_index_genus %>% 
  filter(Taxa %in% ListNoSing(Net_18S_2017_index_genus),
         Taxa %!in% ListNoSing(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by 18S")

#Combine and plot
bind_rows(microscopy_and_DNA, only_microscopy, only_COI, only_18S) %>% 
  mutate(Type = factor(Type, levels = c(
    "Microscopy and DNA", "Only by Microscopy", "Only by COI", "Only by 18S"
  ))) %>%
  filter(RA > 0.001) %>%  ##### NOTE: REMOVED LOW COUNTS! #######
  ggplot() +
  geom_bar(aes(reorder(Taxa, -RA), RA), stat = "identity") +
  theme_minimal_hgrid(9) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,
                                   face = "italic")) +
  facet_wrap("Type", scales = "free_x") +
  ylab("Relative abundance") +
  xlab("Genus")

ggsave("Figure_output/Remaining_Taxa.pdf")
```


```{r}

only_18S %>% 
  summarize(RA = sum(RA))
```

## Conclusions

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

5.  Out of the biomass not detected by the molecular markers, larger taxa, such as amphipods, has the highest biomass.

6.  eDNA of COI and 18S catches many different metazooan taxa not targeted by microscopy. Most of these taxa are benthic mollusks, worms, echinoderms or cnidarians.


# Selection of taxa.

To allow for 

```{r}
# At Genus Level

GenusList <-
  Net_count_2017_index_genus %>%
  
  # 1. Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  
  # 3. Pull list
  pull(Taxa) %>% unique()

GenusList27 <-
  Net_count_2017_index_genus %>%
  
  # 1. Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  
  # 2. Select the 27 most abundant taxa
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>% 
  arrange(-RA) %>% 
  mutate(count = 1:36) %>% 
  filter(count < 28) %>%
  
  # 3. Pull list
  pull(Taxa) %>% unique()

GenusList15 <-
  Net_count_2017_index_genus %>%
  
  # 1. Select taxa present in both microscopy and DNA NET samples
  filter(Taxa %in% ListNoSing(Net_count_2017_index_genus, "m"),
         Taxa %in% c(ListNoSing(Net_18S_2017_index_genus),
                     ListNoSing(Net_COI_2017_index_genus))) %>%
  
  # 2. Select the 27 most abundant taxa
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>% 
  arrange(-RA) %>% 
  mutate(count = 1:36) %>% 
  filter(count < 16) %>%
  
  # 3. Pull list
  pull(Taxa) %>% unique()




GenusList

# At Species Level
SpeciesList <-
  Net_count_2017_index_species %>% 
  filter(Taxa %in% ListNoSing(., "m"),
         Taxa %in% c(ListNoSing(Net_COI_2017_index_species),
                     ListNoSing(eDNA_COI_2017_index_species))) %>% 
  pull(Taxa) %>% unique()
```





# 4. Performance of normalization methods


Next we create a merged dataset for the different markers:

```{r}
Net_merged_data <-
  # Combine data sets
  bind_rows(
    mutate(Net_count_2017_index_genus, marker = "Microscopy"),
    mutate(Net_COI_2017_index_genus, marker = "COI"),
    mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  # Calculate pressence / Absence from the relative abundance
  mutate(PA = ifelse(RA>0.001, 1, 0)) %>% 
  # Remove unwanted taxa and columns
  filter(Taxa %in% GenusList) %>%
  select(Taxa, Date, Abundance, RA, RA_Index, PA, marker) %>% 
  
  # Wrangle the data set to make tidy for comparisons between Molecular/Microscopy
  pivot_longer(3:6, names_to = "Index_Type", values_to = "value") %>% 
  pivot_wider(names_from = marker, values_from = value) %>% 
  pivot_longer(5:6, names_to = "Gene_Marker", values_to = "Molecular") %>% 
  select(Taxa, Date, Gene_Marker, Index_Type, Microscopy, Molecular) %>%
  
  # Remove samples with no matches Molecular/Mic
  filter(is.na(Molecular) == FALSE,
         is.na(Microscopy) == FALSE)



```



## 4.1 Multivariate comparison

### Define functions for Multivariate analyses

```{r}
NMDS <- function(data = Net_merged_data, Marker = "COI", Type = "RA") {
  
  # Construct community matrix dataframe 
  wide <- filter(data, Gene_Marker == Marker, Index_Type == Type) %>%
    select(Taxa, Date, Microscopy, Molecular) %>%
    pivot_longer(3:4, names_to = "Method", values_to = "Value") %>% 
    pivot_wider(names_from = Taxa, values_from = Value)
  
  ## MANTEL TEST
  # Matrix for molecular
  molmat <- wide %>% 
    filter(Method == "Molecular") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Matrix for microscopy 
  micmat <- wide %>% 
    filter(Method == "Microscopy") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Control that sample names are the same
  control <- ifelse(row.names(molmat) != row.names(micmat),"WARNING", "OK")
  ifelse("WARNING" %in% control, print("WARNING"), print("OK"))
  
  # Execute the mantel test
  m.mod <- mantel(vegdist(molmat), vegdist(micmat))
  
  ## PERMANOVA and NMDS
  # Combined community matrix
  mat <- wide %>% 
    mutate(Sample = paste(Date, Method, sep = "_")) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Execute permanova
  permanova <- adonis2(mat ~ Method+Date,
                       data = mutate(wide, Date = format(as.Date(Date),
                                                         format="%m")),
                       permutations = 999, method="bray")
  
  # Execute NMDS
  mod <- metaMDS(mat, trymax = 100)
  
  
  # DATA EXTRACTION
  
  # Extract score coordinates from NMDS
  NMDS_scores <- function(mod) {
    Samples <- 
    scores(mod)$sites %>% 
    as.data.frame() %>%
    rownames_to_column("Name") %>% 
    separate(Name, into = c("Date", "Method"), sep = "_") %>% 
    mutate(Layer = "Samples")

  Taxa <-
    scores(mod)$species %>% 
    as.data.frame() %>%
    rownames_to_column("Taxa") %>% 
    mutate(Layer = "Taxa")

  bind_rows(Samples, Taxa)
  } # End NMDS Scores
  

  out <- NMDS_scores(mod) %>% 
    mutate(Gene_Marker = Marker, Index_Type = Type)
  
  # Combine data set of statistical output
  stats <- tibble(Marker = Marker,
       Type = Type,
       NMDS.stress = mod$stress,
       Method.R2 = permanova$R2[1],
       Method.F = permanova$F[1],
       Method.P = permanova$`Pr(>F)`[1],
       Date.R2 = permanova$R2[2],
       Date.F = permanova$F[2],
       Date.P = permanova$`Pr(>F)`[2],
       Mantel.R = m.mod$statistic,
       Mantel.P = m.mod$signif)
  
  output <- list(NMDS.scores = out,
                 NMDS.stressplot = stressplot(mod),
                 Stats = stats)
  
  
  return(output)
}

```

### Execute all NMDS

```{r}
mod1 <- NMDS(Marker = "COI", Type = "RA")
mod2 <- NMDS(Marker = "COI", Type = "PA")
mod3 <- NMDS(Marker = "COI", Type = "RA_Index")
mod4 <- NMDS(Marker = "18S", Type = "RA")
mod5 <- NMDS(Marker = "18S", Type = "PA")
mod6 <- NMDS(Marker = "18S", Type = "RA_Index")


bind_rows(
  mod1$Stats, mod2$Stats, mod3$Stats, mod4$Stats, mod5$Stats, mod6$Stats) %>% 
  write_csv(file = "./Figure_output/multivariate_stats.csv")

```



```{r}
seasonal_colors = c("#053061", "#3288BD", "#66C2A5", "#7FBC41",
                    "#A6D96A", "#FEE08B", "#FDAE61", "#F46D43",
                    "#D53E4F", "#9E0142", "#67001F", "#40004B")



bind_rows(mod1$NMDS.scores, mod2$NMDS.scores, mod3$NMDS.scores,
          mod4$NMDS.scores, mod5$NMDS.scores, mod6$NMDS.scores) %>% 
  pivot_wider(names_from = Layer,
              values_from = c(NMDS1, NMDS2)) %>%
  mutate(Month = format(as.Date(Date), format="%m")) %>%
  mutate(Index_Type = ifelse(Index_Type == "RA_Index", "eDNA Index", Index_Type)) %>% 
  mutate(Index_Type = factor(Index_Type, levels = c("RA", "PA", "eDNA Index"))) %>% 
  mutate(Taxa = ifelse(Taxa %in% GenusList10, Taxa, "")) %>% 
  
  ggplot(aes(NMDS1_Samples, NMDS2_Samples)) +
  geom_text(aes(NMDS1_Taxa, NMDS2_Taxa, label = Taxa),
            size = 3, colour = "grey", fontface = "italic") +
  geom_line(aes(color = Month, group = Date), linetype = "dotted") +
  geom_point(aes(shape = Method, color = Month), size = 2) +
  stat_ellipse(aes(linetype = Method), size = 0.2) +
  facet_grid(Index_Type~Gene_Marker, scales = "free") +
  theme_minimal_grid(12) +
  theme(aspect.ratio = 1,
        panel.border = element_rect(colour = "black", fill=NA),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  scale_color_manual(values = seasonal_colors) +
  scale_shape_manual(values=c(1, 16))

ggsave("Figure_output/NMDS_both.pdf", width = 8, height = 10)




```

### Permanova

## 4.2 Regression analyses

### Plot correlations

This part is still to be developed. Correlations between the different data sets. One hypothesis is that the correlation is dependent on sequencing depth for different species.

Both 18S and COI combined

```{r}
Net_merged_data %>% 
  filter(Index_Type %in% c("RA_Index", "RA")) %>%
  ggplot(aes(Microscopy, Molecular)) +
  geom_point() +
  geom_smooth(aes(group = Taxa), color = "grey", method = lm, se=F) +
  geom_smooth(color = "#984ea3", method = lm, se=F) +
  theme_minimal_grid() +
  facet_wrap("Index_Type", scales = "free") +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey')) +
  ylim(0,1) +
  xlim(0,1)


ggsave("Figure_output/regression_COI.pdf", width = 7, height = 4)
```

Calculate Rsq, P and Slope for all regressions:

```{r}
#Create new tables for regression summary and calculate coefficients 
species_list <- data.frame(Taxa = unique(Net_merged_data$Taxa),
                           RA_Rsq = 0,
                           #I_Rsq = 0,
                           
                           RA_p.val = 0,
                           #I_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           
                           RA_slope = 0,
                           I_slope = 0)


for(i in 1:length(species_list$Taxa)){ #Start of loop
  RA <- Net_merged_data %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA")
  RA_I <- Net_merged_data %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA_Index")
  
  lm1 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA)
  lm2 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA_I)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  #species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  #species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]

}

species_list %>% 
  write_csv("./Figure_output/regressions.csv")
```

```{r}
species_list %>%
  filter(RA_p.val<0.05) %>% 
  select(RA_slope, I_slope) %>%
  pivot_longer(1:2) %>%
  mutate(Method = factor(name, levels = c("RA_slope", "I_slope")),
         `Slope (Molecular ~ Microscopy)` = value) %>%
  ggplot(aes(Method, `Slope (Molecular ~ Microscopy)`)) +
  geom_violin() +
  geom_point() +
  theme_minimal_hgrid() +
  scale_y_log10() +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey'))

ggsave("./Figure_output/slope_differences.pdf", width = 4, height = 5)
```

# 5. Temporal comparison of taxonomic markers

The aim of this section is to compare if seasonal patterns of RA_index overlap between the different markers and methods. Zooplankton net tows (Count, COI and 18S) that are depth integrated (only one sample per sampling date) are plotted differently from the eDNA samples (that has a depth resolution with several samples per sampling date).\
Genus level comparison:

RA_Index of zooplankton net samples, analysed with the three different markers:

```{r}
tmp <- bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) 

order <- tmp %>% 
  group_by(Taxa, marker) %>% 
  summarize(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>% 
  arrange(-Count) %>% 
  pull(Taxa)

#Combine the three data sets
tmp %>% 
  
  # Filter relevant taxa:
  filter(Taxa %in% GenusList27) %>%
  mutate(marker = factor(marker, levels = c("Count", "COI", "18S"))) %>%
  mutate(Taxa = factor(Taxa, levels = c(order))) %>% 
  
  # Plot
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker, linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  theme_minimal_hgrid() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5) +
  scale_color_manual(values = c('#fec44f','#4daf4a','#377eb8'))

ggsave("Figure_output/Seasonal_Bulk_Genus.pdf", width = 8, height = 10)



```




RA_index of eDNA samples, the average of 18S and COI.

```{r}
# Combine and merge COI and 18S eDNA components. Calculate mean of COI and 18S
eDNA <- bind_rows(mutate(eDNA_COI_2017_index_genus, marker = "COI"),
                  mutate(eDNA_18S_2017_index_genus, marker = "18S",
                         Depth = as.character(Depth))) %>% 
  
  group_by(Taxa, Date, Depth) %>% 
  summarize(RA_Index = mean(RA_Index),
            Abundance = mean(Abundance),
            marker = "eDNA")

# Combine eDNA data with Net data components.
data <-  eDNA %>%
  bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
             mutate(Net_COI_2017_index_genus, marker = "COI"),
             mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 

  filter(Taxa %in% GenusList,
         Taxa %in% pull(eDNA, Taxa)) %>% 
  
  # Modify the parameter shown on the categorical Y axis. That is depth for eDNA samples and Marker for Net samples". 
  mutate(y = ifelse(marker == "eDNA", Depth, marker)) %>% 
  mutate(y = factor(y, levels = rev(c("Count", "COI", "18S", "0", "5", "30", "100", "260")))) %>% 
  mutate(Taxa = factor(Taxa, levels = order))


data %>%  ggplot() +
  
  # Add filter water data:
  geom_point(
    mapping = aes(Date, y, color = RA_Index),
    #data = filter(data, y %!in% c("Count", "COI", "18S")),
    shape = 15) +
  scale_color_gradient2(
    high = '#984ea3', low = "white") +

  # Add Microscopy data
  new_scale_color() +
  geom_point(
    mapping = aes(Date, y, color = RA_Index),
    data = filter(data, y %in% c("Count")),
    shape = 15) +
  scale_color_gradient2(high = '#fec44f',
                       low = "white") +

  # Add Net COI data:
  new_scale_color() +
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("COI")),
    shape = 15) +
  scale_color_gradient2(high = '#4daf4a',
                      low = "white") +
  
  # Add Net 18S data:
  new_scale_color() +
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("18S")),
    shape = 15) +
  scale_color_gradient2(high = '#377eb8',
                       low = "white") +
  
  
  # Set plot themes
  theme_map() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5) +
  facet_wrap("Taxa", ncol = 3)

ggsave("Figure_output/Seasonal_Comb_Genus_press.pdf", width = 7, height = 10)


  


```

## Species level comparison

```{r}

tmp <- bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
          mutate(Net_COI_2017_index_species, marker = "COI_bulk"),
          mutate(EtOH_COI_2017_index_species, marker = "COI_sup"))

order <- tmp %>% 
  group_by(Taxa, marker) %>% 
  summarize(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>% 
  arrange(-Count) %>% 
  pull(Taxa)

tmp %>%
  filter(Taxa %in% SpeciesList) %>%
  mutate(marker = factor(marker, levels = c("Count", "COI_bulk", "COI_sup"))) %>%
  mutate(Taxa = factor(Taxa, levels = order)) %>% 
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker,  linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  theme_minimal_hgrid(8) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        #legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5,
        legend.position = "none") +
  scale_color_manual(values = c('#fec44f','#4daf4a',"cyan"))

ggsave("Figure_output/Seasonal_Bulk_Species.pdf", width = 10, height = 7)
```


# Data amount

```{r}
  # Combine data sets
  bind_rows(
    mutate(Net_count_2017_index_genus, marker = "Microscopy"),
    mutate(Net_COI_2017_index_genus, marker = "COI"),
    mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  # Calculate pressence / Absence from the relative abundance
  # Remove unwanted taxa and columns
  filter(Taxa %in% GenusList) %>%
  select(Taxa, Date, Abundance, marker) %>%
  group_by(Taxa, marker) %>%
  summarise(Abundance = sum(Abundance)) %>% 
  pivot_wider(names_from = "marker", values_from = "Abundance") %>%
  arrange(-Microscopy) %>% 
  write_csv("Figure_output/data_ammount.csv")
```

