---
title: "Comparing Methods"
---

# \*\*\*\*\*\*\*\*\* DATA PREPARATION \*\*\*\*\*\*\*\*\*

Load necessary packages and import functions.

```{r}
# Data import
library(googledrive)
library(readxl)
library(phyloseq)
source("Code/Import_datasets.R")

# All data format modifications and filtering and plotting
library(tidyverse)
'%!in%' <- function(x,y)!('%in%'(x,y))
source("Code/Functions.R")

# Specific plotting tools
#library(ggVennDiagram)
#library(UpSetR)
#library(eulerr)
#library(ggupset)
library(plotly)
library(ggpubr)
library(ggnewscale)
library(paletteer)
library(viridis)
library(cowplot)

# Statistical analyses
library(rjags)
library(zoib)
library(vegan)
```


The flowing code will download the needed data sets from Google Drive server and put them in the Data_import directory. Only has to be run once after cloning this directory.

```{r eval=FALSE, include=FALSE}
# Confirm authorisation with googledrive
drive_auth()

# Update datasets.
# These functions refere to code specifyed under Code/Import_datasets.R.
update_DNA_datasets()
update_ZP_datasets()

```

# 1. Read and combine sequence data sets

## COI data components

The COI Sequencing data are downloaded in the compact format with a separate ASV, Taxonomy, and Sample Data table. The data tables are read separately and then merged into a tidy data tibble, resembling the data format of most zooplankton count data sets.

```{r}
# Import downloaded rds (compact r data) files
ASV_COI <- readRDS("Data_import/COI/ASV_COI.rds")
Samptab_COI <- readRDS("Data_import/COI/Samptab_COI.rds")
Taxtab_COI <- readRDS("Data_import/COI/Taxtab_COI.rds")

# Combine the three components to a tidy data tibble
df_COI <-
  full_join(Taxtab_COI, ASV_COI, by = "ASV") %>%
  pivot_longer(!1:12, names_to = "Library_ID", values_to = "Abundance") %>%
  full_join(Samptab_COI, by = "Library_ID")

df_COI
```

## List contributing Hakai staff 
```{r}
personel <- Samptab_COI %>%
  group_by(Sample_date) %>% 
  summarize(Sample_technician = unique(Sample_technician),
            Extraction_staff = unique(Extraction_staff)) %>%
  pivot_longer(2:3, names_to = "Task", values_to = "technician") %>% 
  separate(technician, sep = ",", into = c("A", "B", "C", "D")) %>% 
  pivot_longer(3:6, values_to = "tecnichian") %>%
  filter(is.na(tecnichian) == FALSE) %>% 
  select(!name)

# Sampling staff  
personel %>% 
  filter(Task == "Sample_technician") %>%
  group_by(tecnichian) %>% 
  summarise(Days_Sampled = n()) %>% 
  arrange(-Days_Sampled)

# Extraction staff
personel %>% 
  filter(Task == "Extraction_staff") %>%
  group_by(tecnichian) %>% 
  summarise(Days_Extracted = n()) %>% 
  arrange(-Days_Extracted)
```


## 18S data components

The 18S Sequencing data are downloaded in the compact format with a separate ASV, Taxonomy, and Sample Data table. The data tables are read separately and then merged into a tidy data tibble, resembling the data format of most zooplankton count data sets.

```{r}
# Import downloaded rds (compact r data) files
ASV_18S <- readRDS("Data_import/18S/ASV_Taxtab_18S.rds")
Samptab_18S <- readRDS("Data_import/18S/Samptab_18S.rds")

# Combine the three components to a tidy data tibble
df_18S <- ASV_18S %>% 
  pivot_longer(!1:8, names_to = "Library_ID", values_to = "Abundance") %>% 
  full_join(Samptab_18S, by = "Library_ID")

df_18S
```

## Microscopy data

The count data set from QU39, counted mainly by Moira.

```{r}
df_count_2017 <- readRDS("Data_import/Microscopy/Hakai_Zooplankton_2013-2017_AN.rds") %>% 
  mutate(Name = Taxon) %>% 
  separate(Taxon, sep= " ", into=c("Genus", "Tax1", "Tax2", "Tax3", "Tax4"))
```

## Clean up
```{r}
rm(ASV_18S, ASV_COI, Samptab_18S, Samptab_COI, Taxtab_COI)
```


# 2. Subset and calculate eDNA indexes of data sets:

In this section the different data formats of the 2017 QU39 time series will be mutated into similar data shapes and transformed in such a way that the data types can be compared. To keep filtering parameters equal for all data types, a set of functions are first defined, that are then executed for all data sets. Comparisons are done both at species and at genus level.

## Define Metazoa

Specifying global filtering parameter: What phyla are present in metazoa?
```{r}
metazoa <- c("Arthropoda",
             "Cnidaria",
             "Chaetognatha",
             "Annelida",
             "Porifera",
             "Echinodermata",
             "Mollusca",
             "Phoronida",
             "Bryozoa",
             "Nemertea",
             "Ctenophora",
             "Priapulida",
             "Rotifera",
             "Chordata",
             "Platyhelminthes")
```

In the following section the functions defined above are used to calculate the relative abundance indexes for the different datasets in the QU39 2017 time series.

### Net Microscopy

Microscopy count data from integrated zooplankton net tows:
```{r}
# Species
Net_count_2017_index_species <- df_count_2017 %>% 
  filter(Station %in% c("QU39"),
         year(Date) == 2017) %>%
  mutate(Depth = DEPTH_STRT,
         Taxa = paste(Genus, Tax1, sep = " "),
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)

# Genus
Net_count_2017_index_genus <- df_count_2017 %>% 
  filter(Station %in% c("QU39"),
         year(Date) == 2017) %>%
  mutate(Depth = DEPTH_STRT,
         Taxa = Genus,
         Abundance = Biomass,
         Sample = Key) %>%
  index_RA(Sample, Taxa, Abundance, Depth, Date)
```

### Net 18S

18S sequence data from integrated zooplankton net tows:
```{r}
# Species
Net_18S_2017_index_species <- df_18S %>% 
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Class %in% metazoa,
         Sample_type == "Zp_Net") %>%
  mutate(Taxa = str_replace(Species, "_", " "),
         Depth = as.character(Line_out_depth),
         Date = Sample_date,
         Sample = Library_ID) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
Net_18S_2017_index_species

# Genus
Net_18S_2017_index_genus <- df_18S %>% 
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Class %in% metazoa,
         Sample_type == "Zp_Net") %>%
  mutate(Taxa = Genus,
         Depth = as.character(Line_out_depth),
         Date = Sample_date,
         Sample = Library_ID) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
Net_18S_2017_index_genus
```

### eDNA 18S

18S sequence data from discrete eDNA Niskin bottles.
```{r}
# Species
eDNA_18S_2017_index_species <- df_18S %>% 
  filter(year(Sample_date) == 2017,
         Site_ID == "QU39",
         Class %in% metazoa) %>%
  mutate(Taxa = str_replace(Species, "_", " "),
         Depth = as.character(Line_out_depth),
         Date = Sample_date,
         Sample = Library_ID) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
eDNA_18S_2017_index_species

# Genus
eDNA_18S_2017_index_genus <- df_18S %>% 
  filter(year(Sample_date) == 2017,
         Site_ID == "QU39",
         Class %in% metazoa) %>%
  mutate(Taxa = Genus,
         Depth = as.character(Line_out_depth),
         Date = Sample_date,
         Sample = Library_ID) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
eDNA_18S_2017_index_genus

```

### Net COI

COI sequence data from integrated zooplankton net tows:
```{r}
# Species
Net_COI_2017_index_species <- df_COI %>%
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Sample_type == "Zp_Net",
         phylum %in% metazoa) %>% 
  mutate(Sample = Library_ID,
         Taxa = species,
         Abundance = Abundance,
         Depth = as.character(Line_out_depth),
         Date = Sample_date) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
Net_COI_2017_index_species

# Genus
Net_COI_2017_index_genus <- df_COI %>%
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Sample_type == "Zp_Net",
         phylum %in% metazoa) %>% 
  mutate(Sample = Library_ID,
         Taxa = genus,
         Abundance = Abundance,
         Depth = as.character(Line_out_depth),
         Date = Sample_date) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
Net_COI_2017_index_genus

```

### eDNA COI

18S sequence data from discrete eDNA Niskin bottles.
```{r}
# Species
eDNA_COI_2017_index_species <- df_COI %>%
  filter(Project_name == "QU39",
         Sample_type == "eDNA",
         year(Sample_date)==2017,
         phylum %in% metazoa) %>% 
  mutate(Sample = Library_ID,
         Taxa = species,
         Abundance = Abundance,
         Depth = as.character(Line_out_depth),
         Date = Sample_date) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
eDNA_COI_2017_index_species

# Genus
eDNA_COI_2017_index_genus <- df_COI %>%
  filter(Project_name == "QU39",
         Sample_type == "eDNA",
         year(Sample_date)==2017,
         phylum %in% metazoa) %>% 
  mutate(Sample = Library_ID,
         Taxa = genus,
         Abundance = Abundance,
         Depth = as.character(Line_out_depth),
         Date = Sample_date) %>% 
  index_RA(Sample, Taxa, Abundance, Depth, Date)
eDNA_COI_2017_index_genus
```





# \*\*\*\*\*\*\*\*\* RESULTS OUTPUT \*\*\*\*\*\*\*\*\*

# 3. Coverage comparison of taxonomic markers

## 3.1 Check library size ect

```{r}


summary_18S <- df_18S %>% 
  filter(Site_ID == "QU39",
         year(Sample_date) == 2017) %>% 
  mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>% 
  mutate(Taxa = ifelse(Class %in% metazoa, "Metazoa", "Other")) %>%
  mutate(Sampling_depth = ifelse(Project_name == "Vertical", "230-0", Line_out_depth)) %>% 
  group_by(Sample_date, Sampling_depth, Taxa, Library_ID) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  group_by(Sample_date, Sampling_depth, Library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Taxa, Abundance) %>%
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>0) %>% 
  mutate(Marker = "18S")

summary_COI <- df_COI %>% 
  filter(Site_ID == "QU39",
         year(Sample_date) == 2017) %>% 
  mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>% 
  mutate(Taxa = ifelse(phylum %in% metazoa, "Metazoa", "Other")) %>%
  mutate(Sampling_depth = ifelse(Project_name == "Vertical", "230-0", Line_out_depth)) %>%
  filter(is.na(Sampling_depth)==FALSE) %>% 
  group_by(Sample_date, Sampling_depth, Taxa, Library_ID) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  group_by(Sample_date, Sampling_depth, Library_ID) %>% 
  reframe(RA = Abundance/sum(Abundance), Taxa, Abundance) %>%
  group_by(Library_ID) %>% 
  filter(sum(Abundance)>0) %>% 
  mutate(Marker = "COI")
  

bind_rows(summary_18S, summary_COI) %>%
  filter(Taxa == "Metazoa") %>%
  mutate(Sampling_depth = factor(Sampling_depth,
                                   levels= c("0", "5", "30",
                                             "100", "260", "230-0"))) %>%
  ggplot() +
  geom_boxplot(aes(Sampling_depth, RA, color = Marker)) +
  theme_cowplot(12) +
  theme(aspect.ratio = 1,
        strip.background =element_blank()) +
  scale_color_manual(values = c('#377eb8','#4daf4a'))

```


## 3.2 Pairwise overlap heat maps:

### Combine lists:

To compare the taxonomic coverage, we combine a list of taxa that pass abundance filtering, using the getSpeciesList, and the getGenusList functions defined in the previous chapter. See the function definition for more information about how data types are filtered.

```{r}
SpeciesListComp <- list(
  `Bulk Microscopy` = getSpeciesList(Net_count_2017_index_species, type = "m"),
  `Bulk 18S` = getSpeciesList(Net_18S_2017_index_species),
  `Bulk COI` = getSpeciesList(Net_COI_2017_index_species),
  `Water COI` = getSpeciesList(eDNA_COI_2017_index_species),
  `Water 18S` = getSpeciesList(eDNA_18S_2017_index_species)
)

GenusListComp <- list(
  `Bulk Microscopy` = getGenusList(Net_count_2017_index_genus, type = "m"),
  `Bulk 18S` = getGenusList(Net_18S_2017_index_genus),
  `Bulk COI` = getGenusList(Net_COI_2017_index_genus),
  `Water COI` = getGenusList(eDNA_COI_2017_index_genus),
  `Water 18S` = getGenusList(eDNA_18S_2017_index_genus)
)
```

### Plot

```{r}

PairwiseOverlap <- function(ListComp = SpeciesListComp, level = "Species Level"){
  
  order <- c("Bulk Microscopy", "Bulk COI", "Water COI", "Bulk 18S", "Water 18S")

  x <- fromList(ListComp)

  t(x) %*% as.matrix(x) %>% 
    as.data.frame() %>% 
    rownames_to_column(var = 'group1') %>% 
    pivot_longer(cols = -group1, names_to = 'group2') %>% 
    mutate(group1 = factor(group1, levels = order),
           group2 = factor(group2, levels = rev(order)),
           level = level)
}


overlap_Plot <- bind_rows(PairwiseOverlap(SpeciesListComp, "Species Level"),
          PairwiseOverlap(GenusListComp, "Genus Level"))


overlap_Plot %>% 
  ggplot(aes(group1, group2)) +
  geom_tile(aes(fill = value)) +
  geom_text(aes(label = value)) +
  theme_bw() +
  scale_fill_viridis(option = "D", direction = -1) +
  labs(fill="Overlap") +
  facet_wrap("level") +
  theme_cowplot(12) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x=element_blank(),
        axis.title.y=element_blank(),
        aspect.ratio = 1,
        strip.background =element_blank())
  
ggsave("Figure_output/Pairwise_Overlap.pdf", width = 7, height = 4.5)
```

### Conclusions:

-   At species level, COI appears to Identify conciderably more of the counted zooplankton taxa than 18S.

-   At genus level all methods align better. COI preforms better than 18S, and Net samples preform better than eDNA.

## 3.3 Proportion of counted biomass detected by eDNA

From the Euler diagram it appears as only a fraction of counted zooplankton is detected by the DNA methods. However, the diagram values all taxa equally and does not account for the relative importance of the taxa. The following code extracts the biomass proportion of counted taxa that are also detected by COI and 18S respectively.

### Prepare data

At Species Level:

```{r}
Sum_per_species <- Net_count_2017_index_species %>% 
  filter(Taxa %in% getSpeciesList(Net_count_2017_index_species, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_species %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_species %>% 
  filter(Taxa %in% getSpeciesList(Net_COI_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")

Detected_18S <- Sum_per_species %>% 
  filter(Taxa %in% getSpeciesList(Net_18S_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_species %>% 
  filter(Taxa %in% c(getSpeciesList(Net_18S_2017_index_species),
                     getSpeciesList(Net_COI_2017_index_species))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

#Plot yearly average
fraction_species <- 
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Species level")
```

At Genus Level:

```{r}
{ ##Species level biomass comparison##
Sum_per_genus <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_genus %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_genus %>% 
  filter(Taxa %in% getGenusList(Net_COI_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")
  
Detected_18S <- Sum_per_genus %>% 
  filter(Taxa %in% getGenusList(Net_18S_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_genus %>% 
  filter(Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

fraction_genus <-   
  Sum_per_day_all %>% 
  mutate(`Bulk COI` = Detected_COI / Counted_Biomass,
         `Bulk 18S` = Detected_18S / Counted_Biomass,
         `Bulk COI & 18S` = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, `Bulk COI`, `Bulk 18S`, `Bulk COI & 18S`) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Genus level")
}


```

### Plot

```{r}
# Combine and plot
Biomass_fraction_plot <- bind_rows(fraction_genus, fraction_species) %>%
  #mutate(Level = factor(Level, levels = c("Species level", "Genus level"))) %>% 
  ggplot(aes(Marker, Proportion)) +
  geom_violin(aes(color = Marker)) +
  cowplot::theme_minimal_hgrid(12) +
  facet_wrap("Level") +
  theme(aspect.ratio = 1,
        axis.title.x=element_blank())
Biomass_fraction_plot

ggsave("Figure_output/Biomass_fractions.pdf", width = 7, height = 4.5)

```


### Conclusions:

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

## 3.4. Exclusive taxa

So the DNA methods combined targets 50-70% of zooplankton my biomass. Here we investigate the remaining genera, that remains un-targeted by the DNA methods. The Genera are ordered by relative biomass contribution.


```{r}
Net_count_2017_index_genus %>% 
  group_by(Taxa) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  mutate(Abundance = Abundance/sum(Abundance))



# Detected by BOTH microscopy and ny of the other methods
microscopy_and_DNA <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(., "m"),
         Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus),
                     getGenusList(eDNA_18S_2017_index_genus),
                     getGenusList(eDNA_COI_2017_index_genus))) %>% 
  #group_by(Taxa, Date) %>% 
  #summarize(RA = sum(RA)) %>% 
  group_by(Taxa) %>% 
  summarise(Abundance = sum(Abundance)) %>% 
  mutate(RA = Abundance/sum(Abundance)) %>% 
  filter(RA > 0.005) %>% 
  mutate(Type = "Microscopy and DNA")

microscopy_and_DNA %>%  pull(RA) %>% sum 
microscopy_and_DNA

# Only detected by microscopy
only_microscopy <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(Net_count_2017_index_genus, type = "m"),
         Taxa %!in% getGenusList(Net_18S_2017_index_genus),
         Taxa %!in% getGenusList(Net_COI_2017_index_genus)) %>% 
  group_by(Taxa, Date) %>% 
  summarize(RA = mean(RA)) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  #reframe(RA = Biomass/sum(Biomass), Taxa) %>% 
  filter(RA > 0.005) %>% 
  mutate(Type = "Only by Microscopy")


# Only detected by COI
only_COI <- eDNA_COI_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(eDNA_COI_2017_index_genus),
         Taxa %!in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by COI")

# Only detected by 18S
only_18S <- eDNA_18S_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(eDNA_18S_2017_index_genus),
         Taxa %!in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by 18S")

#Combine and plot
bind_rows(microscopy_and_DNA, only_microscopy, only_COI, only_18S) %>%
  mutate(Type = factor(Type, levels = c(
    "Microscopy and DNA", "Only by Microscopy", "Only by COI", "Only by 18S"
  ))) %>% 
  ggplot() +
  geom_bar(aes(reorder(Taxa, -RA), RA), stat = "identity") +
  theme_minimal_hgrid(12) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1,
                                   face = "italic")) +
  facet_wrap("Type", scales = "free") +
  ylab("Relative abundance") +
  xlab("Genus")

ggsave("Figure_output/Remaining_Taxa.pdf")
```

## Conclusions

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

5.  Out of the biomass not detected by the molecular markers, larger taxa, such as amphipods, has the highest biomass.

6.  eDNA of COI and 18S catches many different metazooan taxa not targeted by microscopy. Most of these taxa are benthic mollusks, worms, echinoderms or cnidarians.

# 4. Performance of normalization methods

To allow for a data set that can be compared between the different markers, we create a list of all taxa that are detected by both microscopy AND at least one of the genetic methods:

```{r}
# At Genus Level
GenusList <-
  Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(., "m"),
         Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus),
                     getGenusList(eDNA_18S_2017_index_genus),
                     getGenusList(eDNA_COI_2017_index_genus))) %>% 
  pull(Taxa) %>% unique()

# At Species Level
SpeciesList <-
  Net_count_2017_index_species %>% 
  filter(Taxa %in% getSpeciesList(., "m"),
         Taxa %in% c(getSpeciesList(Net_COI_2017_index_species),
                     getSpeciesList(eDNA_COI_2017_index_species))) %>% 
  pull(Taxa) %>% unique()
```

Next we create a merged dataset for the different markers:

```{r}
Net_merged_data <-
  # Combine data sets
  bind_rows(
    mutate(Net_count_2017_index_genus, marker = "Microscopy"),
    mutate(Net_COI_2017_index_genus, marker = "COI"),
    mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  # Calculate pressence / Absence from the relative abundance
  mutate(PA = ifelse(RA>0.001, 1, 0)) %>% 
  # Remove unwanted taxa and columns
  filter(Taxa %in% GenusList) %>%
  select(Taxa, Date, Abundance, RA, RA_Index, PA, marker) %>% 
  
  # Wrangle the data set to make tidy for comparisons between Molecular/Microscopy
  pivot_longer(3:6, names_to = "Index_Type", values_to = "value") %>% 
  pivot_wider(names_from = marker, values_from = value) %>% 
  pivot_longer(5:6, names_to = "Gene_Marker", values_to = "Molecular") %>% 
  select(Taxa, Date, Gene_Marker, Index_Type, Microscopy, Molecular) %>%
  
  # Remove samples with no matches Molecular/Mic
  filter(is.na(Molecular) == FALSE,
         is.na(Microscopy) == FALSE)

Net_merged_data


```

## 4.1 Multivariate comparison

### Define functions for Multivariate analyses

```{r}
NMDS <- function(data = Net_merged_data, Marker = "COI", Type = "RA") {
  
  # Construct community matrix dataframe 
  wide <- filter(data, Gene_Marker == Marker, Index_Type == Type) %>%
    select(Taxa, Date, Microscopy, Molecular) %>%
    pivot_longer(3:4, names_to = "Method", values_to = "Value") %>% 
    pivot_wider(names_from = Taxa, values_from = Value)
  
  ## MANTEL TEST
  # Matrix for molecular
  molmat <- wide %>% 
    filter(Method == "Molecular") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Matrix for microscopy 
  micmat <- wide %>% 
    filter(Method == "Microscopy") %>% 
    mutate(Sample = Date) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Control that sample names are the same
  control <- ifelse(row.names(molmat) != row.names(micmat),"WARNING", "OK")
  ifelse("WARNING" %in% control, print("WARNING"), print("OK"))
  
  # Execute the mantel test
  m.mod <- mantel(vegdist(molmat), vegdist(micmat))
  
  ## PERMANOVA and NMDS
  # Combined community matrix
  mat <- wide %>% 
    mutate(Sample = paste(Date, Method, sep = "_")) %>%
    select(!1:2) %>% 
    column_to_rownames("Sample") %>% 
    as.matrix()
  
  # Execute permanova
  permanova <- adonis2(mat ~ Method+Date,
                       data = mutate(wide, Date = format(as.Date(Date),
                                                         format="%m")),
                       permutations = 999, method="bray")
  
  # Execute NMDS
  mod <- metaMDS(mat, trymax = 100)
  
  
  # DATA EXTRACTION
  
  # Extract score coordinates from NMDS
  NMDS_scores <- function(mod) {
    Samples <- 
    scores(mod)$sites %>% 
    as.data.frame() %>%
    rownames_to_column("Name") %>% 
    separate(Name, into = c("Date", "Method"), sep = "_") %>% 
    mutate(Layer = "Samples")

  Taxa <-
    scores(mod)$species %>% 
    as.data.frame() %>%
    rownames_to_column("Taxa") %>% 
    mutate(Layer = "Taxa")

  bind_rows(Samples, Taxa)
  } # End NMDS Scores
  

  out <- NMDS_scores(mod) %>% 
    mutate(Gene_Marker = Marker, Index_Type = Type)
  
  # Combine data set of statistical output
  stats <- tibble(Marker = Marker,
       Type = Type,
       NMDS.stress = mod$stress,
       Method.R2 = permanova$R2[1],
       Method.F = permanova$F[1],
       Method.P = permanova$`Pr(>F)`[1],
       Date.R2 = permanova$R2[2],
       Date.F = permanova$F[2],
       Date.P = permanova$`Pr(>F)`[2],
       Mantel.R = m.mod$statistic,
       Mantel.P = m.mod$signif)
  
  output <- list(NMDS.scores = out,
                 NMDS.stressplot = stressplot(mod),
                 Stats = stats)
  
  
  return(output)
}

```

### Execute all NMDS

```{r}
mod1 <- NMDS(Marker = "COI", Type = "RA")
mod2 <- NMDS(Marker = "COI", Type = "PA")
mod3 <- NMDS(Marker = "COI", Type = "RA_Index")
mod4 <- NMDS(Marker = "18S", Type = "RA")
mod5 <- NMDS(Marker = "18S", Type = "PA")
mod6 <- NMDS(Marker = "18S", Type = "RA_Index")


bind_rows(
  mod1$Stats, mod2$Stats, mod3$Stats, mod4$Stats, mod5$Stats, mod6$Stats) %>% 
  write_csv(file = "./Figure_output/multivariate_stats.csv")

```

COI

```{r}
seasonal_colors = c("#053061", "#3288BD", "#66C2A5", "#7FBC41",
                    "#A6D96A", "#FEE08B", "#FDAE61", "#F46D43",
                    "#D53E4F", "#9E0142", "#67001F", "#40004B")



bind_rows(mod1$NMDS.scores, mod2$NMDS.scores, mod3$NMDS.scores) %>% 
  pivot_wider(names_from = Layer,
              values_from = c(NMDS1, NMDS2)) %>%
  mutate(Month = format(as.Date(Date), format="%m")) %>%
  mutate(Index_Type = ifelse(Index_Type == "RA_Index", "eDNA Index", Index_Type)) %>% 
  mutate(Index_Type = factor(Index_Type, levels = c("RA", "PA", "eDNA Index"))) %>% 
  
  ggplot(aes(NMDS1_Samples, NMDS2_Samples)) +
  geom_text(aes(NMDS1_Taxa*1.1, NMDS2_Taxa*1.1, label = Taxa),
            size = 3, colour = "grey", fontface = "italic") +
  geom_line(aes(color = Month, group = Date), linetype = "dotted") +
  geom_point(aes(shape = Method, color = Month), size = 3) +
  facet_wrap("Index_Type", scales = "free", ncol = 2) +
  theme_minimal_grid(12) +
  theme(aspect.ratio = 1) +
  scale_color_manual(values = seasonal_colors)

ggsave("Figure_output/NMDS_COI.pdf", width = 8, height = 7)

```

18S
```{r}
seasonal_colors = c("#053061", "#3288BD", "#66C2A5", "#7FBC41",
                    "#A6D96A", "#FEE08B", "#FDAE61", "#F46D43",
                    "#D53E4F", "#9E0142", "#67001F", "#40004B")



bind_rows(mod1$NMDS.scores, mod2$NMDS.scores, mod3$NMDS.scores,
          mod4$NMDS.scores, mod5$NMDS.scores, mod6$NMDS.scores) %>% 
  pivot_wider(names_from = Layer,
              values_from = c(NMDS1, NMDS2)) %>%
  mutate(Month = format(as.Date(Date), format="%m")) %>%
  mutate(Index_Type = ifelse(Index_Type == "RA_Index", "eDNA Index", Index_Type)) %>% 
  mutate(Index_Type = factor(Index_Type, levels = c("RA", "PA", "eDNA Index"))) %>% 
  
  ggplot(aes(NMDS1_Samples, NMDS2_Samples)) +
  geom_text(aes(NMDS1_Taxa*1.1, NMDS2_Taxa*1.1, label = Taxa),
            size = 3, colour = "grey", fontface = "italic") +
  geom_line(aes(color = Month, group = Date), linetype = "dotted") +
  geom_point(aes(shape = Method, color = Month), size = 2) +
  facet_grid(Index_Type~Gene_Marker, scales = "free") +
  theme_minimal_grid(12) +
  theme(aspect.ratio = 1,
        panel.border = element_rect(colour = "black", fill=NA),
        legend.position="none") +
  scale_color_manual(values = seasonal_colors)

ggsave("Figure_output/NMDS_both.pdf", width = 8, height = 10)




```





### Permanova

## 4.2 Regression analyses

### Plot correlations

This part is still to be developed. Correlations between the different data sets. One hypothesis is that the correlation is dependent on sequencing depth for different species.

Both 18S and COI combined

```{r}
Net_merged_data %>% 
  filter(Index_Type %in% c("RA_Index", "RA")) %>%
  ggplot(aes(Microscopy, Molecular)) +
  geom_point() +
  geom_smooth(aes(group = Taxa), color = "grey", method = lm, se=F) +
  geom_smooth(color = "#984ea3", method = lm, se=F) +
  theme_minimal_grid() +
  facet_wrap("Index_Type", scales = "free") +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey'))


ggsave("Figure_output/regression_COI.pdf", width = 7, height = 4)
```

Calculate Rsq, P and Slope for all regressions:

```{r}
#Create new tables for regression summary and calculate coefficients 
species_list <- data.frame(Taxa = unique(Net_merged_data$Taxa),
                           RA_Rsq = 0,
                           #I_Rsq = 0,
                           
                           RA_p.val = 0,
                           #I_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           
                           RA_slope = 0,
                           I_slope = 0)


for(i in 1:length(species_list$Taxa)){ #Start of loop
  RA <- Net_merged_data %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA")
  RA_I <- Net_merged_data %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA_Index")
  
  lm1 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA)
  lm2 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA_I)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  #species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  #species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]

}

species_list %>% 
  write_csv("./Figure_output/regressions.csv")
```

```{r}
species_list %>%
  filter(RA_p.val<0.05) %>% 
  select(RA_slope, I_slope) %>%
  pivot_longer(1:2) %>%
  mutate(Method = factor(name, levels = c("RA_slope", "I_slope")),
         `Slope (Molecular ~ Microscopy)` = value) %>%
  ggplot(aes(Method, `Slope (Molecular ~ Microscopy)`)) +
  geom_violin() +
  geom_point() +
  theme_minimal_hgrid() +
  scale_y_log10() +
  theme(aspect.ratio = 1,
        panel.background = element_rect(colour = 'darkgrey'))

ggsave("./Figure_output/slope_differences.pdf", width = 4, height = 5)
```

## 4.3 Slope correction - new try

```{r}
Net_merged_data_COI <- Net_merged_data %>% 
  filter(Gene_Marker == "COI")


#Create new tables for regression summary and calculate coefficients 
species_list <- data.frame(Taxa = unique(Net_merged_data_COI$Taxa),
                           RA_Rsq = 0,
                           #I_Rsq = 0,
                           
                           RA_p.val = 0,
                           #I_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           
                           RA_slope = 0,
                           I_slope = 0)


for(i in 1:length(species_list$Taxa)){ #Start of loop
  RA <- Net_merged_data_COI %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA")
  RA_I <- Net_merged_data_COI %>% 
    filter(Taxa == species_list$Taxa[i],
           Index_Type == "RA_Index")
  
  lm1 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA)
  lm2 <- lm(sqrt(Microscopy) ~ sqrt(Molecular), data = RA_I)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  #species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  #species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]

}

species_list
```

```{r}
Net_merged_data_COI


# Calculate slopes and maximum relative contribution per Taxa
adj_COI <- Net_merged_data_COI %>%
  select(Taxa, Date, Index_Type, Molecular) %>% 
  pivot_wider(names_from = Index_Type, values_from = Molecular) %>% 
  mutate(RA_adj = Abundance) %>% 
  group_by(Date) %>% 
  mutate(Total_Abundance = sum(Abundance))
adj_COI
slopes <- adj_COI %>% 
  group_by(Taxa)  %>%
  summarise(max_contribution = max(RA)) %>% 
  merge(species_list, by="Taxa")

# Adjust slopes not to be corrected for
slopes
#slopes[slopes$RA_Rsq < 0.3,]$RA_slope <- 1
#slopes[slopes$RA_p > 0.05,]$RA_slope <- 1
#slopes[slopes$RA_slope > 0.6 & slopes$RA_slope < 1,4,]$RA_slope <- 1
slopes[is.na(slopes)]<-1

slopes <- slopes %>% 
  arrange(desc(max_contribution))

slopes



for(j in 1:nrow(slopes)){
  Species <- slopes$Taxa[j]
  Slope <- slopes$RA_slope[j]^2
  adj_COI[adj_COI$Taxa == Species,]$RA_adj <- (Slope*(adj_COI[adj_COI$Taxa == Species,]$Total_Abundance* adj_COI[adj_COI$Taxa == Species,]$RA_adj-adj_COI[adj_COI$Taxa == Species,]$RA_adj^2))/(adj_COI[adj_COI$Taxa == Species,]$Total_Abundance - Slope* adj_COI[adj_COI$Taxa == Species,]$RA_adj)
  adj_COI <- adj_COI %>%
    group_by(Date) %>%
    mutate(Total_Abundance = sum(RA_adj))
}


adj_COI <- adj_COI %>%
  group_by(Date) %>% 
  mutate(RA_adj = RA_adj/sum(RA_adj))


Net_merged_data_COI
adjusted_data <- Net_merged_data_COI %>%
  select(Taxa, Date, Index_Type, Microscopy) %>% 
  pivot_wider(names_from = Index_Type, values_from = Microscopy) %>% 
  left_join(adj_COI, by = c("Taxa", "Date"), suffix = c("_Microscopy", "_COI"))

adjusted_data


```

```{r}
adjusted_data %>% 
  ggplot(aes(RA_Microscopy, RA_COI)) +
  geom_point() +
  geom_smooth(aes(group = Taxa), color = "grey", method = lm, se=F) +
  geom_smooth(color = "red", method = lm, se=F)
```

```{r}
adjusted_data %>% 
  ggplot(aes(RA_Microscopy, RA_adj)) +
  geom_point() +
  geom_smooth(aes(group = Taxa), color = "grey", method = lm, se=F) +
  geom_smooth(color = "red", method = lm, se=F) +
  ylim(0, 1) +
  xlim(0,1)
  
```

```{r}
tmp <- adjusted_data %>% 
  select(Taxa, Date, RA_Microscopy, RA_adj) %>% 
  pivot_longer(3:4, names_to = "Method", values_to = "Value") %>%
  mutate(Value = ifelse(Value < 0, 0, Value)) %>% 
  transmute(Sample = paste(Date, Method, sep = "_"),
            Taxa, Value) %>% 
  pivot_wider(names_from = Taxa, values_from = Value) %>% 
  column_to_rownames("Sample") %>% 
  as.matrix()

tmp

mod <- metaMDS(tmp, trymax = 100)
stressplot(mod)
print(mod)

NMDS_scores(mod) %>% 
  filter(Layer == "Samples") %>% 
  mutate(Month = format(as.Date(Date), format="%m")) %>% 
  
  ggplot(aes(NMDS1, NMDS2)) +
  geom_line(aes(color = Month, group = Date), linetype = "dotted") +
  geom_point(aes(shape = Method, color = Month), size = 3) +
  theme_bw()

```

```{r}
#Part 2. Adjust seq_count by slopes

#meta_long - MOTU data table in long format with the following columns:
#Station = Station name
#Preservative = ethanol or dry
#Species = Species or taxa name as identified by metabarcoding
#Morph.id = Species or taxa name at the lowest common taxonomic level with microscopy analysis (for example all species belonging to Cirripedia are assigned as Cirripedia)
#seq_count = number of sequence reads in the sample
#percent_count = relative proportion of reads in a sample (0-1)
#Total.reads = total sequence reads in the sample

#Slopes_DW - data table with the following columns, from the previous step:
#Morph.id = Species or taxa name at the lowest common taxonomic level with microscopy analysis (for example all species belonging to Cirripedia are assigned as Cirripedia)
#slope = slope of the regression line between square-root transformed proportion biomass and square-root transformed proportion sequence reads
#Rsq = strength of the regression
#p.val = significance of the regression

#calculate maximum contribution of each species/taxa
contr <- meta_long %>%
  group_by(Morph.id)  %>%
  summarise(max_contribution = max(percent_count))

#merge with slopes table
slopes1 <- merge(Slopes, contr, by="Morph.id")

#do not adjust taxa (slope=1) where the following conditions are not met:
#R2<=0.3, p<=0.05, slope<=0.6 OR slope>=1.4

slopes1[slopes1$Rsq > 0.3,]$slope <- 1
slopes1[slopes1$p.sig > 0.05,]$slope <- 1
slopes1[slopes1$slope > 0.6 & slopes1$slope < 1,4,]$slope <- 1
slopes1[is.na(slopes1)]<-1

#Sort taxa in descending order by contribution
slopes1 <- slopes %>% 
  arrange(desc(max_contribution))

#Make new column seq_count_adj, by default equal to seq_count (no adjustment)
meta_long$seq_count_adj <- meta_long$seq_count

#Apply conversion factors one by one and recalculate Total.reads

for(j in 1:nrow(slopes1)){
  Species <- slopes1$Morph.id[j]
  Slope <- slopes1$slope[j]^2
  meta_long[meta_long$Morph.id == Species,]$seq_count_adj <- (Slope*(meta_long[meta_long$Morph.id == Species,]$Total.reads* meta_long[meta_long $Morph.id == Species,]$seq_count_adj-meta_long[meta_long$Morph.id == Species,]$seq_count_adj^2))/(meta_long[meta_long$Morph.id == Species,]$Total.reads - Slope* meta_long[meta_long$Morph.id == Species,]$seq_count_adj)
  meta_long <- meta_long %>%
    group_by(Station, Preservative) %>%
    mutate(Total.reads = sum(seq_count_adj))
}
```

# 5. Temporal comparison of taxonomic markers

The aim of this section is to compare if seasonal patterns of RA_index overlap between the different markers and methods. Zooplankton net tows (Count, COI and 18S) that are depth integrated (only one sample per sampling date) are plotted differently from the eDNA samples (that has a depth resolution with several samples per sampling date).\
Genus level comparison:

RA_Index of zooplankton net samples, analysed with the three different markers:

```{r}
#Combine the three data sets
bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  
  # Filter relevant taxa:
  filter(Taxa %in% GenusList) %>%
  mutate(marker = factor(marker, levels = c("Count", "COI", "18S"))) %>% 
  
  # Plot
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker, linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 6) +
  theme_minimal_hgrid() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5) +
  scale_color_manual(values = c('#fec44f','#4daf4a','#377eb8'))

ggsave("Figure_output/Seasonal_Bulk_Genus.pdf", width = 12, height = 7)
```

RA_index of eDNA samples, the average of 18S and COI.

```{r}
# Combine and merge COI and 18S eDNA components. Calculate mean of COI and 18S
eDNA <- bind_rows(mutate(eDNA_COI_2017_index_genus, marker = "COI"),
                  mutate(eDNA_18S_2017_index_genus, marker = "18S",
                         Depth = as.character(Depth))) %>% 
  
  group_by(Taxa, Date, Depth) %>% 
  summarize(RA_Index = mean(RA_Index),
            Abundance = mean(Abundance),
            marker = "eDNA")

# Combine eDNA data with Net data components.
data <-  eDNA %>%
  bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
             mutate(Net_COI_2017_index_genus, marker = "COI"),
             mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 

  filter(Taxa %in% GenusList,
         Taxa %in% pull(eDNA, Taxa)) %>% 
  
  # Modify the parameter shown on the categorical Y axis. That is depth for eDNA samples and Marker for Net samples". 
  mutate(y = ifelse(marker == "eDNA", Depth, marker),
         y = factor(y, levels = rev(c("Count", "COI", "18S", "0", "5", "30", "100", "260"))))
 
 
data %>% ggplot() +
  geom_point(
    aes(Date, y, color = RA_Index),
    #filter(data, y %!in% c("Count", "COI")),
    shape = 15) +
  #facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#984ea3',
                       low = "white") +
  
  new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("Count")),
    shape = 15) +
  #facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#fec44f',
                       low = "white") +
  
    new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("COI")),
    shape = 15) +
  #facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#4daf4a',
                       low = "white") +
  
      new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("18S")),
    shape = 15) +
  facet_wrap(facets = "Taxa", ncol = 5) +
  scale_color_gradient2(high = '#377eb8',
                       low = "white") +
  
  theme_map() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5)

ggsave("Figure_output/Seasonal_Comb_Genus_press.pdf", width = 7, height = 10)




```

## Species level comparison

```{r}

bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
          mutate(Net_COI_2017_index_species, marker = "COI")) %>% 
  filter(Taxa %in% SpeciesList) %>%
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker,  linetype = marker)) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  theme_minimal_hgrid(10) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        #legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5,
        legend.position = "none") +
  scale_color_manual(values = c('#4daf4a','#fec44f'))

ggsave("Figure_output/Seasonal_Bulk_Species.pdf", width = 10, height = 7)
```

## Comnined plot species level

```{r}
data <- bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
                  mutate(Net_COI_2017_index_species, marker = "COI"),
                  mutate(eDNA_COI_2017_index_species, marker = "eDNA"),
                  tibble(marker = "_", Date = ymd("2017-01-01"),
                         RA_Index = 0, Taxa = "Acartia longiremis")) %>% 
  filter(Taxa %in% SpeciesList,
         Taxa %in% pull(eDNA_COI_2017_index_species, Taxa)) %>% 
  mutate(y = ifelse(marker == "eDNA", Depth, marker),
         y = factor(y, levels = rev(c("Count", "COI", "_", "0", "5", "30", "100", "260"))))


data %>% ggplot() +
  geom_point(
    aes(Date, y, color = RA_Index),
    #filter(data, y %!in% c("Count", "COI")),
    shape = 15) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#4daf4a',
                       low = "white") +
  
  new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("Count")),
    shape = 15) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#fec44f',
                       low = "white") +
  
    new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("COI")),
    shape = 15) +
  facet_wrap(facets = "Taxa", ncol = 3) +
  scale_color_gradient2(high = '#4daf4a',
                       low = "white") +
  
  theme_map() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        strip.text = element_text(face = "italic"),
        legend.position="none",
        panel.background = element_rect(fill = '#F5F5F5', colour = 'darkgrey'),
        aspect.ratio = 1/2.5)

ggsave("Figure_output/Seasonal_comb_Species.pdf", width = 7, height = 10)

```


