---
title: "Comparing Methods"
---

# 0. Prepare environment

Load necessary packages and import functions.

```{r}
# Data import
library(googledrive)
library(readxl)
library(phyloseq)
source("Code/Import_datasets.R")

# All data format modifications and filtering and plotting
library(tidyverse)
'%!in%' <- function(x,y)!('%in%'(x,y))

# Specific plotting tools
library(ggVennDiagram)
library(UpSetR)
library(eulerr)
library(ggupset)
library(plotly)
library(ggpubr)
library(ggnewscale)

# Statistical analyses
library(rjags)
library(zoib)
library(vegan)
```

The flowing code will download the needed data sets from Google Drive server and put them in the Data_import directory. Only has to be run once after cloning this directory.

```{r eval=FALSE, include=FALSE}
# Confirm authorisation with googledrive
drive_auth()

# Update datasets.
# These functions refere to code specifyed under Code/Import_datasets.R.
update_DNA_datasets()
update_ZP_datasets()
update_Contributed_data()

```

# 1. Read and combine sequence data sets

## COI data components

The COI Sequencing data are downloaded in the compact format with a separate ASV, Taxonomy, and Sample Data table. The data tables are read separately and then merged into a tidy data tibble, resembling the data format of most zooplankton count data sets.

```{r}
# Import downloaded rds (compact r data) files
ASV_COI <- readRDS("Data_import/COI/ASV_COI.rds")
Samptab_COI <- readRDS("Data_import/COI/Samptab_COI.rds")
Taxtab_COI <- readRDS("Data_import/COI/Taxtab_COI.rds")

# Combine the three components to a tidy data tibble
df_COI <-
  full_join(Taxtab_COI, ASV_COI, by = "ASV") %>%
  pivot_longer(!1:12, names_to = "Library_ID", values_to = "Abundance") %>%
  full_join(Samptab_COI, by = "Library_ID")
```

## 18S data components

The 18S Sequencing data are downloaded in the compact format with a separate ASV, Taxonomy, and Sample Data table. The data tables are read separately and then merged into a tidy data tibble, resembling the data format of most zooplankton count data sets.

```{r}
# Import downloaded rds (compact r data) files
ASV_18S <- readRDS("Data_import/18S/ASV_18S.rds")
Samptab_18S <- readRDS("Data_import/18S/Samptab_18S.rds")
Taxtab_18S <- readRDS("Data_import/18S/Taxtab_18S.rds")

# Combine the three components to a tidy data tibble
df_18S <- 
  full_join(Taxtab_18S, ASV_18S, by = "Sequence") %>% 
  pivot_longer(!1:10, names_to = "Library_ID", values_to = "Abundance") %>% 
  full_join(Samptab_18S, by = "Library_ID") %>% 
  filter(is.na(ASV)==FALSE)
```

This is the contributed data set from Catherinas 18S sequencing project. The data is downloaded as a Phyloseq object, and melted into a data frame to resemble the data above.

```{r}
df_18S_ts <- readRDS("Data_import/Contributed18S/ps_18S.rds") %>% psmelt()
```

## Microscopy data

Count data from the vertical aspect of Zoopsprint 2022. Counted mainly by Christie.

```{r}
df_count_vertical <- read_excel("Data_import/Microscopy/Vertical_ZP_2022.xlsx")
 
```

The count data set from QU39, counted mainly by Moira.

```{r}
df_count_2017 <- readRDS("Data_import/Microscopy/Hakai_Zooplankton_2013-2017_AN.rds") %>% 
  mutate(Name = Taxon) %>% 
  separate(Taxon, sep= " ", into=c("Genus", "Tax1", "Tax2", "Tax3", "Tax4"))
```

## Defining metazoan phyla

Specifying global filtering parameter: What phyla are present in metazoa?

```{r}
metazoa <- c("Arthropoda",
             "Cnidaria",
             "Chaetognatha",
             "Annelida",
             "Porifera",
             "Echinodermata",
             "Mollusca",
             "Phoronida",
             "Bryozoa",
             "Nemertea",
             "Ctenophora",
             "Priapulida",
             "Rotifera",
             "Chordata",
             "Platyhelminthes")
```

# 2. Subset and calculate eDNA indexes of data sets:

In this section the different data formats of the 2017 QU39 time series will be mutated into similar data shapes and transformed in such a way that the data types can be compared. To keep filtering parameters equal for all data types, a set of functions are first defined, that are then executed for all data sets. Comparisons are done both at species and at genus level.

## Defining transformation and filtering functions

The following function will calculate relative abundance and the index of relative abundance (RA_Index) for any data set. The RA_Index is identical to eDNA Index, but as other data types than eDNA is used here, we choose to give the index a more generic name. The functions are specific for the given analysis workflow and are not meant to be exported.

```{r}

#function to calculate eDNA index
eDNAINDEX <- function(x) { #where x is a dataframe with taxa/OTUs/etc in rows, and samples in columns
  rowMax <- function(x){apply(x, MARGIN = 1, FUN = max)}
  temp <- sweep(x, MARGIN = 2, STATS = colSums(x), FUN = "/") #create proportion for each taxon/OTU within each sample
  sweep(temp, MARGIN = 1, STATS = rowMax(temp), FUN = "/")
}





collapseRAindex <- function(data) {
  
  # This function requires a data frame that contains at least the following parameters:
  # Abundance (read count or taxonomic count)
  # Date (date of sampling)
  # Taxa (can be either genus or species)
  # Depth (a factor, eiter a discrete depth or a depth interval)
  
  output <- data %>%
    # Interpret na as 0
    mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>%
    
    # Summarising abundance per date, depth and taxa.
    group_by(Date, Taxa, Depth) %>% 
    summarise(Abundance = sum(Abundance)) %>%
    
    # Calculate relative abundance (RA) for each samples.
    # Contain taxa and original abundance information.
    group_by(Date, Depth) %>% 
    reframe(RA = Abundance/sum(Abundance),
            Taxa, Abundance) %>%
    filter(is.na(RA) == FALSE) %>%
    
        
    #Remove outliers before calculating Index:
    group_by(Taxa) %>%
    #filter(RA <= quantile(RA, 0.99)) %>% 
    reframe(RA = ifelse(RA <= quantile(RA, 0.99), RA, -1),
            Date, Depth, Abundance) %>%
    group_by(Taxa) %>%
    reframe(RA = ifelse(RA == -1, max(RA), RA),
            Date, Depth, Abundance) %>% 
    
    # Calculate Index for each taxa.
    # Contain depth, date, Abundance and RA information.
    group_by(Taxa) %>%
    reframe(Date, Depth, Abundance, RA,
            RA_Index = RA/max(RA)) %>% 
    filter(is.na(RA_Index) == FALSE)

  return(output)
}
```

The following function will generate a list of prevalent species by filtering the output generated by the function above. The function will remove species with inadequate species names, and remove species with generally low occurrences throughout the data set.

```{r}
getSpeciesList <- function(data, type = "DNA") {
  # This function require the output of "collapseRAindex" function.
  
  # Prefiltration for DNA data.
  # An initial filtering step will remov any low observation.
  if (type == "DNA") {data <- filter(data, Abundance > 50)}
  
  List <- data %>% 
  
    # Remove low occurrences.
    # Each species has to be observed in more than 2 samples.
    filter(RA_Index > 0) %>% 
    group_by(Taxa) %>%
    filter(n() > 2) %>% 
    
    # Remove bad species names:
    # The following name patterns indicate that species level identification failed. 
    filter(!grepl('sp.', Taxa),
           !grepl('_sp', Taxa),
           !grepl(' sp', Taxa),
           grepl(' ', Taxa),
           is.na(Taxa)==FALSE,
           Taxa != "no identification") %>%
    
    # Special case database name correction:
    mutate(Taxa = ifelse(Taxa== "Discoconchoecia aff. elegans CMARA05309_Os109.1.1",
                          "Discoconchoecia elegans", Taxa)) %>% 
  
    # Get Taxa names
    pull(Taxa) %>% unique()

  return(List)  
  # This function returns a name vector of species that passed filtering parameters.
  
}
```

The following function will generate a list of prevalent genera by filtering the output generated by the function above. The function will remove genera with inadequate genus names, and remove genera with generally low occurrences throughout the data set.

```{r}
getGenusList <- function(data, type="DNA") {
  # This function require the output of "collapseRAindex" function.
  
  # Prefiltration for DNA data.
  # An initial filtering step will remove any low observation.
  if (type == "DNA") {data <- filter(data, Abundance > 50)}
  
  List <- data %>%
  
    # Remove low occurrences
    # Each genus has to be observed in more than 5 samples.
    filter(RA_Index > 0) %>% 
    group_by(Taxa) %>%
    filter(n() > 5) %>%
    
    # Remove bad genus names
    # The following name patterns indicate that genus level identification failed. 
    filter(is.na(Taxa)==FALSE,
           !grepl('_', Taxa),
           Taxa != "no identification",
           Taxa != "unknown genus") %>% 
    # Get Taxa names
    pull(Taxa) %>% unique()

  return(List)  
  # This function returns a name vector of genera that passed filtering parameters.
}
```

## Calculate RA_Index

In the following section the functions defined above are used to calculate the relative abundance indexes for the different datasets in the QU39 2017 time series.

### Net Count

Microscopy count data from integrated zooplankton net tows:

```{r}
# Filter samples in the dataset
Net_count_2017 <- df_count_2017 %>% 
  filter(Station %in% c("QU39"),
         year(Date) == 2017)
  
# Species level calculation  
Net_count_2017_index_species <- Net_count_2017 %>% 
  # Define what columns contaim Taxa, Abundance, Depth, and Date:
  mutate(Taxa = paste(Genus, Tax1, sep = " "),
         Abundance = Biomass,
         Depth= DEPTH_STRT,
         Date = Date) %>% 
  # Calculate RA_index
  collapseRAindex()


# Genus level claculation
Net_count_2017_index_genus <- Net_count_2017 %>% 
  # Define Taxa, abundance, Depth, Date:
  mutate(Taxa = Genus,
         Abundance = Biomass,
         Depth= DEPTH_STRT,
         Date = Date) %>% 
  # Calculate RA index
  collapseRAindex()
```

### Net 18S

18S sequence data from integrated zooplankton net tows:

```{r}
# Filter samples in the dataset
Net_18S_2017 <- df_18S %>% 
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Division == "Metazoa",
         Sample_type == "Zp_Net")

# Species level  
Net_18S_2017_index_species <- Net_18S_2017 %>% 
  # Define Taxa, abundance, Depth Date:
  mutate(Taxa = str_replace(Species, "_", " "),
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()

# Genus level  
Net_18S_2017_index_genus <- Net_18S_2017 %>% 
  # Define Taxa, abundance, Depth Date:
  mutate(Taxa = Genus,
         Abundance = Abundance,
         Depth=Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()
```

### eDNA 18S

18S sequence data from discrete eDNA Niskin bottles.

```{r}
# Filter samples in the dataset
eDNA_18S_2017 <- df_18S_ts %>% 
  filter(year(collected) == 2017,
         site_id == "QU39",
         Kingdom == "Metazoa")

# Species level index
eDNA_18S_2017_index_species <- eDNA_18S_2017 %>% 
  mutate(Taxa = str_replace(Species, "_", " "),
         Abundance = Abundance,
         Depth = line_out_depth,
         Date = date(collected)) %>% 
  collapseRAindex()

# Genus level index
eDNA_18S_2017_index_genus <- eDNA_18S_2017 %>% 
  mutate(Taxa = Genus,
         Abundance = Abundance,
         Depth = line_out_depth,
         Date = date(collected)) %>% 
  collapseRAindex()


```

### Net COI

COI sequence data from integrated zooplankton net tows:

```{r}
# Filter samples in the dataset
Net_COI_2017 <- df_COI %>%
  filter(Project_name == "Vertical",
         year(Sample_date) == 2017,
         Sample_type == "Zp_Net",
         phylum %in% metazoa)

# Species level index
Net_COI_2017_index_species <- Net_COI_2017 %>% 
  mutate(Taxa = species,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()

# Species level index
Net_COI_2017_index_genus <- Net_COI_2017 %>% 
  mutate(Taxa = genus,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()
```

### eDNA COI

18S sequence data from discrete eDNA Niskin bottles.

```{r}
# Filter samples in the dataset
eDNA_COI_2017 <- df_COI %>%
  filter(Project_name == "QU39",
         Sample_type == "eDNA",
         year(Sample_date)==2017,
         phylum %in% metazoa)

# Species level index
eDNA_COI_2017_index_species <- eDNA_COI_2017 %>% 
  mutate(Taxa = species,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()

# Species level index
eDNA_COI_2017_index_genus <- eDNA_COI_2017 %>% 
  mutate(Taxa = genus,
         Abundance = Abundance,
         Depth = Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindex()
```

# 3. Coverage comparison of taxonomic markers

## Combine lists:

To compare the taxonomic coverage, we combine a list of taxa that pass abundance filtering, using the getSpeciesList, and the getGenusList functions defined in the previous chapter. See the function definition for more information about how data types are filtered.

```{r}
SpeciesListComp <- list(
  `Net Microscopy` = getSpeciesList(Net_count_2017_index_species, type = "m"),
  `Net 18S` = getSpeciesList(Net_18S_2017_index_species),
  `Net COI` = getSpeciesList(Net_COI_2017_index_species),
  `eDNA COI` = getSpeciesList(eDNA_COI_2017_index_species),
  `eDNA 18S` = getSpeciesList(eDNA_18S_2017_index_species)
)

GenusListComp <- list(
  `Net Microscopy` = getGenusList(Net_count_2017_index_genus, type = "m"),
  `Net 18S` = getGenusList(Net_18S_2017_index_genus),
  `Net COI` = getGenusList(Net_COI_2017_index_genus),
  `eDNA COI` = getGenusList(eDNA_COI_2017_index_genus),
  `eDNA 18S` = getGenusList(eDNA_18S_2017_index_genus)
)
```

## Pairwise overlap heat maps:

```{r}

PairwiseOverlap <- function(ListComp = SpeciesListComp, level = "Species Level"){
  
  order <- c("Net Microscopy", "Net COI", "eDNA COI", "Net 18S", "eDNA 18S")

  x <- fromList(ListComp)

  t(x) %*% as.matrix(x) %>% 
    as.data.frame() %>% 
    rownames_to_column(var = 'group1') %>% 
    pivot_longer(cols = -group1, names_to = 'group2') %>% 
    mutate(group1 = factor(group1, levels = order),
           group2 = factor(group2, levels = rev(order)),
           level = level)
}



bind_rows(PairwiseOverlap(SpeciesListComp, "Species Level"),
          PairwiseOverlap(GenusListComp, "Genus Level")) %>% 
  ggplot(aes(group1, group2)) +
  geom_tile(aes(fill = value)) +
  geom_text(aes(label = value)) +
  theme_classic() +
  scale_fill_gradient2(high = scales::muted("red"),
                       low = scales::muted("white")) +
  facet_wrap("level") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        axis.title.x=element_blank(),
        axis.title.y=element_blank())
  
  
ggsave("Figure_output/Pairwise_Overlap.pdf", width = 7, height = 4.5)
```

## Upset plots

## Upset plots are a way to study comparisons in overlaps.

```{r}
# Species Level
upset(fromList(SpeciesListComp),
      sets = c("Net Microscopy", "Net 18S", "Net COI", "eDNA 18S", "eDNA COI"))

# Genus Level
upset(fromList(GenusListComp),
      sets = c("Net Microscopy", "Net 18S", "Net COI", "eDNA COI", "eDNA 18S"))
```

## Euler diagrams

The same information can be shown in Euler diagrams.

```{r}

  par(mfcol = c(1,2))
  set.seed(104)
  SpeciesListComp %>% 
    fromList() %>% 
    euler(shape = "ellipse") %>% 
    plot(quantities = TRUE,
         fill = "transparent",
         lty = 1:5)

  set.seed(102)
  GenusListComp %>% 
    fromList() %>% 
    euler(shape = "ellipse") %>% 
    plot(quantities = TRUE,
         fill = "transparent",
         lty = 1:5)

```

**Conclusions:**

-   At species level, COI appears to Identify conciderably more of the counted zooplankton taxa than 18S.

-   At genus level all methods align better. COI preforms better than 18S, and Net samples preform better than eDNA.

The same code repeated but exported as PDF file:

```{r}
pdf("Figure_output/Euler_species.pdf")
set.seed(104)
SpeciesListComp %>% 
  fromList() %>% 
  euler(shape = "ellipse") %>% 
  plot(quantities = TRUE,
       fill = "transparent",
        lty = 1:5)
dev.off()

pdf("Figure_output/Euler_genus.pdf")
set.seed(102)
GenusListComp %>% 
  fromList() %>% 
  euler(shape = "ellipse") %>% 
  plot(quantities = TRUE,
       fill = "transparent",
       lty = 1:5)
dev.off()

```

## Proportion of counted biomass detected by eDNA

From the Euler diagram it appears as only a fraction of counted zooplankton is detected by the DNA methods. However, the diagram values all taxa equally and does not account for the relative importance of the taxa. The following code extracts the biomass proportion of counted taxa that are also detected by COI and 18S respectively.

At Species Level:

```{r}
Sum_per_species <- Net_count_2017_index_species %>% 
  filter(Taxa %in% getSpeciesList(Net_count_2017_index_species, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_species %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_species %>% 
  filter(Taxa %in% getSpeciesList(Net_COI_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")

Detected_18S <- Sum_per_species %>% 
  filter(Taxa %in% getSpeciesList(Net_18S_2017_index_species)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_species %>% 
  filter(Taxa %in% c(getSpeciesList(Net_18S_2017_index_species),
                     getSpeciesList(Net_COI_2017_index_species))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

#Plot yearly average
fraction_species <- 
  Sum_per_day_all %>% 
  mutate(Net_COI = Detected_COI / Counted_Biomass,
         Net_18S = Detected_18S / Counted_Biomass,
         Net_COI_and_18S = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, Net_COI, Net_18S, Net_COI_and_18S) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Species level")
```

At Genus Level:

```{r}
{ ##Species level biomass comparison##
Sum_per_genus <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance))

Sum_per_day <- Sum_per_genus %>%
  group_by(Date) %>% 
  summarize(Counted_Biomass = sum(Biomass))

Detected_COI <- Sum_per_genus %>% 
  filter(Taxa %in% getGenusList(Net_COI_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_COI = sum(Biomass)) %>% 
  left_join(Sum_per_day, by = "Date")
  
Detected_18S <- Sum_per_genus %>% 
  filter(Taxa %in% getGenusList(Net_18S_2017_index_genus)) %>% 
  group_by(Date) %>% 
  summarize(Detected_18S = sum(Biomass)) %>% 
  left_join(Detected_COI, by = "Date")

Sum_per_day_all <- Sum_per_genus %>% 
  filter(Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus))) %>% 
  group_by(Date) %>%
  summarize(Detected_18S_COI = sum(Biomass)) %>% 
  left_join(Detected_18S, by = "Date")

fraction_genus <-   
  Sum_per_day_all %>% 
  mutate(Net_COI = Detected_COI / Counted_Biomass,
         Net_18S = Detected_18S / Counted_Biomass,
         Net_COI_and_18S = Detected_18S_COI /Counted_Biomass) %>% 
  select(Date, Net_COI, Net_18S, Net_COI_and_18S) %>% 
  pivot_longer(2:4, names_to = "Marker", values_to = "Proportion") %>% 
  mutate(Level = "Genus level")}
```

Results combined and plotted:

```{r}
# Combine and plot
bind_rows(fraction_genus, fraction_species) %>%
  mutate(Level = factor(Level, levels = c("Species level", "Genus level"))) %>% 
  ggplot(aes(Marker, Proportion)) +
  geom_violin() +
  theme_classic() +
  facet_wrap("Level") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
  
ggsave("Figure_output/Biomass_fractions.pdf", width = 7, height = 4.5)
```

**Conclusions:**

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

## Taxa NOT detected by DNA vs Microscopy:

So the DNA methods combined targets 50-70% of zooplankton my biomass. Here we investigate the remaining genera, that remains un-targeted by the DNA methods. The Genera are ordered by relative biomass contribution.

```{r}
# Detected by BOTH microscopy and ny of the other methods
microscopy_and_DNA <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(., "m"),
         Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus),
                     getGenusList(eDNA_18S_2017_index_genus),
                     getGenusList(eDNA_COI_2017_index_genus))) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance)) %>% 
  group_by(Taxa) %>% 
  summarize(Biomass = mean(Biomass)) %>%
    reframe(RA = Biomass/sum(Biomass),
          Taxa) %>% 
  filter(RA > 0.005) %>% 
  mutate(Type = "Microscopy and DNA")

# Only detected by microscopy
only_microscopy <- Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(Net_count_2017_index_genus, type = "m"),
         Taxa %!in% getGenusList(Net_18S_2017_index_genus),
         Taxa %!in% getGenusList(Net_COI_2017_index_genus)) %>% 
  group_by(Taxa, Date) %>% 
  summarize(Biomass = sum(Abundance)) %>% 
  group_by(Taxa) %>% 
  summarize(Biomass = mean(Biomass)) %>%
  reframe(RA = Biomass/sum(Biomass),
          Taxa) %>% 
  filter(RA > 0.005) %>% 
  mutate(Type = "Only by Microscopy")

# Only detected by COI
only_COI <- eDNA_COI_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(eDNA_COI_2017_index_genus),
         Taxa %!in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by COI")

# Only detected by 18S
only_18S <- eDNA_18S_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(eDNA_18S_2017_index_genus),
         Taxa %!in% getGenusList(Net_count_2017_index_genus, type = "m")) %>% 
  group_by(Taxa) %>% 
  summarize(RA = mean(RA)) %>%
  mutate(Type = "Only by 18S")

#Combine and plot
bind_rows(microscopy_and_DNA, only_microscopy, only_COI, only_18S) %>%
  mutate(Type = factor(Type, levels = c(
    "Microscopy and DNA", "Only by Microscopy", "Only by COI", "Only by 18S"
  ))) %>% 
  ggplot() +
  geom_bar(aes(reorder(Taxa, -RA), RA), stat = "identity") +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  facet_wrap("Type", scales = "free") +
  ylab("Relatice abundance") +
  xlab("Genus")

ggsave("Figure_output/Remaining_Taxa.pdf")
```

## Conclusions:

1.  18S Cannot be used to identify taxa at species level, the gene is to conserved.

2.  COI Covers around 38-75% of the ZP-net biomass at the specie level.

3.  All markers have higher agreement on the genus level, but COI still outperforms 18S also at the genus level.

4.  Combining 18S and COI is slightly, but only marginally better than using COI alone.

5.  Out of the biomass not detected by the molecular markers, larger taxa, such as amphipods, has the highest biomass.

6.  eDNA of COI and 18S catches many different metazooan taxa not targeted by microscopy. Most of these taxa are benthic mollusks, worms, echinoderms or cnidarians.

# 4. Temporal comparison of taxonomic markers

The aim of this section is to compare if seasonal patterns of RA_index overlap between the different markers and methods. Zooplankton net tows (Count, COI and 18S) that are depth integrated (only one sample per sampling date) are plotted differently from the eDNA samples (that has a depth resolution with several samples per sampling date).\
\
Before sampling we construct lists of species and genera that are observed both with microscopy AND at least one of the molecular markers:

```{r}
GenusList <-
  Net_count_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(., "m"),
         Taxa %in% c(getGenusList(Net_18S_2017_index_genus),
                     getGenusList(Net_COI_2017_index_genus),
                     getGenusList(eDNA_18S_2017_index_genus),
                     getGenusList(eDNA_COI_2017_index_genus))) %>% 
  pull(Taxa) %>% unique()


SpeciesList <-
  Net_count_2017_index_species %>% 
  filter(Taxa %in% getSpeciesList(., "m"),
         Taxa %in% c(getSpeciesList(Net_COI_2017_index_species),
                     getSpeciesList(eDNA_COI_2017_index_species))) %>% 
  pull(Taxa) %>% unique()
```

## Genus level comparison:

RA_Index of zooplankton net samples, analysed with the three different markers:

```{r}
#Combine the three data sets
bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  
  # Filter relevant taxa:
  filter(Taxa %in% GenusList) %>%
  
  # Plot
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker)) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  labs(title = "Genus Level Zooplankton-Net Marker Comparison") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

ggsave("Figure_output/Seasonal_Net_Genus_Comp.pdf")
```
```{r}
#Combine the three data sets
bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  
  # Filter relevant taxa:
  filter(Taxa %in% GenusList) %>%
  
  # Plot
  ggplot() +
  geom_line(aes(Date, RA, color = marker)) +
  facet_wrap(facets = "Taxa", scales = "free_y") +
  theme_bw() +
  labs(title = "Genus Level Zooplankton-Net Marker Comparison") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

ggsave("Figure_output/Seasonal_Net_Genus_Comp.pdf")
```








RA_index of eDNA samples, the average of 18S and COI.

```{r}

bind_rows(mutate(eDNA_COI_2017_index_genus, marker = "COI"),
          mutate(eDNA_18S_2017_index_genus, marker = "18S",
                 Depth = as.character(Depth))) %>% 
  filter(Taxa %in% GenusList) %>% 
  group_by(Taxa, Date, Depth) %>% 
  summarize(RA_Index = mean(RA_Index),
            Abundance = mean(Abundance)) %>%
  mutate(Depth = factor(Depth, levels = rev(c("0", "5", "30", "100", "260")))) %>% 
  ggplot() +
  
  geom_point(aes(Date, Depth, color = RA_Index), shape=15) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  theme_bw() +
  scale_color_gradient2(high = scales::muted("red"),
                       low = scales::muted("white")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "Genus Level eDNA with Vertical Resolution")

ggsave("Figure_output/Seasonal_eDNA_Genus_Comp.pdf", height = 6)
```

## Comnined plot genus level

```{r}
eDNA <- bind_rows(mutate(eDNA_COI_2017_index_genus, marker = "COI"),
                  mutate(eDNA_18S_2017_index_genus, marker = "18S",
                         Depth = as.character(Depth))) %>% 
  
  group_by(Taxa, Date, Depth) %>% 
  summarize(RA_Index = mean(RA_Index),
            Abundance = mean(Abundance),
            marker = "eDNA")

data <-  eDNA %>%
  bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
             mutate(Net_COI_2017_index_genus, marker = "COI"),
             mutate(Net_18S_2017_index_genus, marker = "18S"),
            tibble(marker = "_", Date = ymd("2017-01-01"),
                         RA_Index = 0, Taxa = "Acartia")) %>% 

  filter(Taxa %in% GenusList,
         Taxa %in% pull(eDNA, Taxa)) %>% 
  
  mutate(y = ifelse(marker == "eDNA", Depth, marker),
         y = factor(y, levels = rev(c("Count", "COI", "18S", "_", "0", "5", "30", "100", "260"))))
 
data
 
data %>% ggplot() +
  geom_point(
    aes(Date, y, color = RA_Index),
    #filter(data, y %!in% c("Count", "COI")),
    shape = 15) +
  facet_wrap(facets = "Taxa") +
  scale_color_gradient2(high = scales::muted("red"),
                       low = scales::muted("white")) +
  
  new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("Count", "COI", "18S")),
    shape = 15) +
  facet_wrap(facets = "Taxa") +
  scale_color_gradient2(high = scales::muted("blue"),
                       low = scales::muted("white")) +
  
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "Genus level temporal comparison")

ggsave("Figure_output/Seasonal_combined_Species_Comp.pdf", height = 6)

```

## Outliers:

```{r}
eDNA_COI_2017_index_genus %>%
  filter(Taxa %in% GenusList,
         RA_Index > 0) %>% 
  ggplot(aes(x = RA_Index)) +
  geom_histogram(bins = 100L) +
  facet_wrap(facets = "Taxa")

Net_COI_2017_index_genus %>%
  filter(Taxa %in% GenusList,
         Abundance > 0) %>%
  group_by(Taxa) %>% 
  summarise(median = median(Abundance),
            mean = mean(Abundance))

eDNA_COI_2017_index_genus
```

## Species level comparison

```{r}

bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
          mutate(Net_COI_2017_index_species, marker = "COI")) %>% 
  filter(Taxa %in% SpeciesList) %>%
  ggplot() +
  geom_line(aes(Date, RA_Index, color = marker)) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  labs(title = "Integrated Zooplankton Nets Species Level") +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

ggsave("Figure_output/Seasonal_Net_Species_Comp.pdf")
```

## Comnined plot species level

```{r}
data <- bind_rows(mutate(Net_count_2017_index_species, marker = "Count"),
                  mutate(Net_COI_2017_index_species, marker = "COI"),
                  mutate(eDNA_COI_2017_index_species, marker = "eDNA"),
                  tibble(marker = "_", Date = ymd("2017-01-01"),
                         RA_Index = 0, Taxa = "Acartia longiremis")) %>% 
  filter(Taxa %in% SpeciesList,
         Taxa %in% pull(eDNA_COI_2017_index_species, Taxa)) %>% 
  mutate(y = ifelse(marker == "eDNA", Depth, marker),
         y = factor(y, levels = rev(c("Count", "COI", "_", "0", "5", "30", "100", "260"))))
  
data %>% ggplot() +
  geom_point(
    aes(Date, y, color = RA_Index),
    #filter(data, y %!in% c("Count", "COI")),
    shape = 15) +
  facet_wrap(facets = "Taxa") +
  scale_color_gradient2(high = scales::muted("red"),
                       low = scales::muted("white")) +
  
  new_scale_color() +
  
  geom_point(
    aes(Date, y, color = RA_Index),
    filter(data, y %in% c("Count", "COI")),
    shape = 15) +
  facet_wrap(facets = "Taxa") +
  scale_color_gradient2(high = scales::muted("blue"),
                       low = scales::muted("white")) +
  
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "Species level temporal comparison")

ggsave("Figure_output/Seasonal_combined_Species_Comp.pdf", height = 6)

```

eDNA (COI only)

```{r}

eDNA_COI_2017_index_species %>% 
  filter(Taxa %in% SpeciesList) %>% 
  group_by(Taxa, Date, Depth) %>% 
  mutate(Depth = factor(Depth, levels = rev(c("0", "5", "30", "100", "260")))) %>% 
  ggplot() +
  
  geom_point(aes(Date, Depth,
                 color = RA_Index), shape = 15) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  theme_bw() +
  scale_color_gradient2(high = scales::muted("red"),
                       low = scales::muted("white")) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "eDNA Vertical at Genus Level")

ggsave("Figure_output/Seasonal_eDNA_Species_Comp.pdf", height = 6)
```

## Genus level regression

```{r}
#Combine the three data sets
bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
          mutate(Net_COI_2017_index_genus, marker = "COI"),
          mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  
  # Filter relevant taxa:
  filter(Taxa %in% GenusList)


```

## Plot correlations

This part is still to be developed. Correlations between the different data sets.\
\
One hypothesis is that the correlation is dependent on sequencing depth for different species.

```{r}
bind_rows(mutate(Net_count_2017_index_genus, marker = "Count"),
            mutate(Net_COI_2017_index_genus, marker = "COI"),
            mutate(Net_18S_2017_index_genus, marker = "18S")) %>% 
  filter(Taxa %in% GenusList)

  
  left_join(
    Net_count_2017_index_genus,
    Net_COI_2017_index_genus,
    by = c("Taxa", "Date"),
    suffix = c(".count", ".COI")) %>%
  filter(Taxa %in% GenusList) %>% 
  ggplot(aes(RA_Index.count, RA_Index.COI)) +
  geom_point() +
  geom_smooth(method = lm) +
  #facet_wrap(facets = "Taxa") +
  theme_bw() +
  ylim(0, 1) +
  facet_wrap("Taxa")
  
 d <- left_join(
    Net_count_2017_index_genus,
    Net_COI_2017_index_genus,
    by = c("Taxa", "Date"),
    suffix = c(".count", ".COI"))
 d
 
 logit <- function(p) { log(p/(1-p)) }
 



  
lm(logit(RA_Index.COI) ~ logit(RA_Index.count) + Taxa, data = d)

zoib(RA_Index.COI ~ RA_Index.count, data = d)

betareg(RA_Index.COI ~ RA_Index.count + Taxa, data = d)
```

## lm count corrections

```{r}
# Compile dataset for comparison:


comp.d <- left_join(
    Net_count_2017_index_genus,
    Net_COI_2017_index_genus,
    by = c("Taxa", "Date"),
    suffix = c(".count", ".COI")) %>%
  filter(Taxa %in% GenusList) %>% 
  mutate(RA.count.sq = RA.count^0.5,
         RA.COI.sq = RA.COI^0.5) %>% 
  filter(is.na(RA.COI.sq) == FALSE,
         is.na(RA.count.sq) == FALSE)
comp.d
  
# Compare original regression slopes with eDNA index regresson slopes:
comp.d %>%
  ggplot() +
  geom_point(aes(RA.count, RA.COI), color = "blue") +
  geom_smooth(aes(RA.count, RA.COI), method = lm, color = "blue") +
  geom_point(aes(RA_Index.count, RA_Index.COI), color = "red") +
  geom_smooth(aes(RA_Index.count, RA_Index.COI), method = lm, color = "red") +
  #facet_wrap(facets = "Taxa") +
  theme_bw() +
  ylim(0, 1) +
  facet_wrap("Taxa") +
  geom_abline(slope = 1)

# Original slopes only.
comp.d %>%
  ggplot() +
  geom_point(aes(RA.count, RA.COI), color = "blue") +
  geom_smooth(aes(RA.count, RA.COI), method = lm, color = "blue") +
  #facet_wrap(facets = "Taxa") +
  theme_bw() +
  ylim(0, 1) +
  facet_wrap("Taxa", scales = "free_y") +
  geom_abline(slope = 1)
```

```{r}
#Create new tables for regression summary and calculate coefficients 
species_list <- data.frame(Taxa = unique(comp.d$Taxa),
                           RA_Rsq = 0,
                           I_Rsq = 0,
                           
                           RA_p.val = 0,
                           I_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           
                           RA_slope = 0,
                           I_slope = 0)
species_list

for(i in 1:length(species_list$Taxa)){ #Start of loop
  data=comp.d[comp.d$Taxa == species_list$Taxa[i],]
  lm1 <- lm(RA.count.sq ~ RA.COI.sq, data = data)
  lm2 <- lm(sqrt(RA_Index.count) ~ sqrt(RA_Index.COI), data = data)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]
}

species_list
```
Full regression

```{r}

lmT <- lm(RA.count.sq ~ RA.COI.sq + Taxa, data = comp.d)
lmT <- lm(sqrt(RA_Index.count) ~ sqrt(RA_Index.COI) + Taxa, data = comp.d)
summary(lmT)


plot(lmT)
comp.d %>% 
  ggplot() +
  geom_point(aes(RA.count, RA.COI), color = "blue") +
  geom_smooth(aes(RA.count, RA.COI), method = lm, color = "blue") +
  geom_point(aes(RA_Index.count, RA_Index.COI), color = "red") +
  geom_smooth(aes(RA_Index.count, RA_Index.COI), method = lm, color = "red") +
  #facet_wrap(facets = "Taxa") +
  theme_bw() +
  ylim(0, 1) +
  geom_abline(slope = 1)
```

The R2 and P values are identical for RA and Index regressions. Only improvement of slopes.

Also, R2 is linearly correlated with the slope....

```{r}
p <- species_list %>% 
  ggplot() +
  geom_point(aes(I_slope, RA_Rsq, color = Taxa))
ggplotly(p)
```

Slope correction

```{r}
slope_corr <- comp.d %>%
  group_by(Taxa) %>% 
  summarise(max_contribution = mean(RA.count)) %>% 
  arrange(desc(max_contribution)) %>% 
  left_join(species_list, by = "Taxa") %>% 
  left_join(comp.d, by = "Taxa") %>% 
  group_by(Date) %>% 
  mutate(Abundance.COI.adj = 
           ifelse(RA_p.val < 0.05 & RA_Rsq > 0.3,
                  ((RA_slope^2)*(sum(Abundance.COI)*Abundance.COI-Abundance.COI^2)) / 
                    (sum(Abundance.COI)-RA_slope^2*Abundance.COI),
                  Abundance.COI)) %>%
  group_by(Date) %>% 
  mutate(RA.COI.adj = Abundance.COI.adj/sum(Abundance.COI.adj))
slope_corr

slope_corr %>% 
  ggplot() +
  geom_point(aes(RA.count, RA.COI.adj), color = "blue") +
  geom_smooth(aes(RA.count, RA.COI.adj), method = lm, color = "blue") +
  
  geom_point(aes(RA.count, RA.COI), color = "red") +
  geom_smooth(aes(RA.count, RA.COI), method = lm, color = "red") +
  
  geom_abline(slope = 1) +
  
  theme_bw() +
  ylim(0, 1) +
  facet_wrap("Taxa", scales = "free_y")

```

```{r}
slope_corr <- comp.d %>%
  group_by(Taxa) %>% 
  summarise(max_contribution = mean(RA.count)) %>% 
  arrange(desc(max_contribution)) %>% 
  left_join(species_list, by = "Taxa") %>% 
  left_join(comp.d, by = "Taxa") %>% 
  group_by(Date) %>% 
  mutate(Total.abundance.COI = sum(Abundance.COI),
         Abundance.COI.adj = Abundance.COI)

Taxa_ordered <- slope_corr %>% 
  pull(Taxa) %>% unique

for (j in 1:length(Taxa_ordered)) {
  
  Taxa_j <- Taxa_ordered[2]
  
  slope_corr <- slope_corr %>% 
    mutate(Abundance.COI.adj = 
           ifelse(RA_p.val < 0.05 & RA_Rsq > 0.3 & Taxa == Taxa_j & between(RA_slope, 0.6, 1.4)==FALSE,
                  (RA_slope^2*(Total.abundance.COI*Abundance.COI.adj-Abundance.COI.adj^2))/ 
                    (Total.abundance.COI-RA_slope^2*Abundance.COI.adj),
                  Abundance.COI.adj)) %>% 
    group_by(Date) %>% 
    mutate(Total.abundance.COI = sum(Abundance.COI.adj))
}

slope_corr



slope_corr %>%
  group_by(Date) %>% 
  mutate(RA.COI.adj = Abundance.COI.adj/sum(Abundance.COI.adj)) %>% 
  ggplot() +
  geom_point(aes(RA.count, RA.COI.adj), color = "blue") +
  geom_smooth(aes(RA.count, RA.COI.adj), method = lm, color = "blue") +
  
  geom_point(aes(RA.count, RA.COI), color = "red") +
  geom_smooth(aes(RA.count, RA.COI), method = lm, color = "red") +
  
  geom_abline(slope = 1) +
  
  theme_bw() +
  ylim(0, 1) +
  facet_wrap("Taxa", scales = "free_y")

```

```{r}
#Create new tables for regression summary and calculate coefficients 
species_list <- data.frame(Taxa = unique(comp.d$Taxa),
                           RA_Rsq = 0,
                           I_Rsq = 0,
                           Adj_Rsq = 0,
                           
                           RA_p.val = 0,
                           I_p.val = 0,
                           Adj_p.val = 0,
                           
                           RA_B0 = 0,
                           I_B0 = 0,
                           Adj_B0 = 0, 
                           
                           RA_slope = 0,
                           I_slope = 0,
                           Adj_slope = 0)
species_list

for(i in 1:length(species_list$Taxa)){ #Start of loop
  data=slope_corr[slope_corr$Taxa == species_list$Taxa[i],]
  lm1 <- lm(RA.count.sq ~ RA.COI.sq, data = data)
  lm2 <- lm(sqrt(RA_Index.count) ~ sqrt(RA_Index.COI), data = data)
  lm3 <- lm(RA.count.sq ~ sqrt(RA.COI.adj), data = data)
  species_list$RA_Rsq[i] <- summary(lm1)$adj.r.squared
  species_list$RA_p.val[i] <- summary(lm1)$coefficients[2,4]
  species_list$RA_B0[i] <- lm1$coefficients[1]
  species_list$RA_slope[i] <- lm1$coefficients[2]
  species_list$I_Rsq[i] <- summary(lm2)$adj.r.squared
  species_list$I_p.val[i] <- summary(lm2)$coefficients[2,4]
  species_list$I_B0[i] <- lm2$coefficients[1]
  species_list$I_slope[i] <- lm2$coefficients[2]
  species_list$Adj_Rsq[i] <- summary(lm3)$adj.r.squared
  species_list$Adj_p.val[i] <- summary(lm3)$coefficients[2,4]
  species_list$Adj_B0[i] <- lm3$coefficients[1]
  species_list$Adj_slope[i] <- lm3$coefficients[2]
}

species_list
```

## NMDS

### eDNA index

```{r}

mat <- comp.d %>% 
  select(Taxa, Date, RA_Index.count, RA_Index.COI) %>% 
  pivot_longer(3:4, names_to = "Marker") %>% 
  mutate(Marker = ifelse(Marker == "RA_Index.count", "Count", "COI"),
         Date = as.character(Date)) %>% 
  mutate(Sample = paste(Date, Marker, sep = "_"), Taxa, value) %>%
  select(Sample, Taxa, value) %>% 
  pivot_wider(names_from = Taxa, values_from = value) %>% 
  column_to_rownames("Sample") %>% 
  as.matrix()

NMDS <- metaMDS(sqrt(mat), trymax = 100)
stressplot(NMDS)
NMDS

Samples <- 
  scores(NMDS)$sites %>% 
  as.data.frame() %>%
  rownames_to_column("Name") %>% 
  separate(Name, into = c("Date", "Marker"), sep = "_")

Taxa <-
  scores(NMDS)$species %>% 
  as.data.frame() %>%
  rownames_to_column("Taxa")

NMDS_Index <-
  Samples %>% 
  ggplot(aes(NMDS1, NMDS2)) +
  geom_line(aes(color = Date), linetype = "dotted") +
  geom_point(aes(shape = Marker, color = Date), size = 3) +
  geom_segment(data = Taxa,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = Taxa, aes(NMDS1*1.1, NMDS2*1.1, label = Taxa)) +
  theme_bw()

```

### RA

```{r}
mat <- comp.d %>% 
  select(Taxa, Date, RA.count, RA.COI) %>% 
  pivot_longer(3:4, names_to = "Marker") %>% 
  mutate(Marker = ifelse(Marker == "RA.count", "Count", "COI"),
         Date = as.character(Date)) %>% 
  #(Marker == "Count") %>% 
  mutate(Sample = paste(Date, Marker, sep = "_"), Taxa, value) %>%
  select(Sample, Taxa, value) %>% 
  pivot_wider(names_from = Taxa, values_from = value) %>% 
  column_to_rownames("Sample") %>% 
  as.matrix()
  mat

NMDS <- metaMDS(sqrt(mat), trymax = 100)
stressplot(NMDS)
NMDS

Samples <- 
  scores(NMDS)$sites %>% 
  as.data.frame() %>%
  rownames_to_column("Name") %>% 
  separate(Name, into = c("Date", "Marker"), sep = "_")

Taxa <-
  scores(NMDS)$species %>% 
  as.data.frame() %>%
  rownames_to_column("Taxa")

NMDS_relative_abundance <-
  Samples %>% 
  ggplot(aes(NMDS1, NMDS2)) +
  geom_line(aes(color = Date), linetype = "dotted") +
  geom_point(aes(shape = Marker, color = Date), size = 3) +
  geom_segment(data = Taxa,
               aes(x = 0, xend = NMDS1, y = 0, yend = NMDS2),
               arrow = arrow(length = unit(0.25, "cm")), colour = "grey") +
  geom_text(data = Taxa, aes(NMDS1*1.1, NMDS2*1.1, label = Taxa)) +
  theme_bw()
```

```{r}
ggarrange(NMDS_relative_abundance, NMDS_Index, common.legend = TRUE)


```



```{r}
#### Count/COI Residual Plot
  
left_join(Net_count_2017_index_genus,
          Net_COI_2017_index_genus,
          by = c("Taxa", "Date"),
          suffix = c(".count", ".COI")) %>%
  filter(Taxa %in% GenusList) %>% 
  mutate(Diff = RA_Index.COI/RA_Index.count) %>% 
  ggplot(aes(Abundance.COI, abs(log10(Diff)))) +
  geom_point()




```

0 means that RA_Index for count and COI is identical. The higher the count number of COI, the closer to 0 is the residuals.

## Experiment: Include all taxa

In this plot, I include all taxa that passes quality filtering for COI on genus level. Revealing a handful of additional species with interesting seasonal patterns (as it appears)

```{r}
eDNA_COI_2017_index_genus %>% 
  filter(Taxa %in% getGenusList(eDNA_COI_2017_index_genus)) %>% 
  group_by(Taxa, Date, Depth) %>% 
  mutate(Depth = factor(Depth, levels = rev(c("0", "5", "30", "100", "260")))) %>% 
  ggplot() +
  
  geom_point(aes(Date, Depth,
                 color = RA_Index, size = RA_Index, alpha = RA_Index)) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  scale_color_viridis_b() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "eDNA Vertical at Genus Level")


eDNA_COI_2017_index_species %>% 
  filter(Taxa %in% getSpeciesList(.)) %>% 
  group_by(Taxa, Date, Depth) %>% 
  mutate(Depth = factor(Depth, levels = rev(c("0", "5", "30", "100", "260")))) %>% 
  ggplot() +
  
  geom_point(aes(Date, Depth,
                 color = RA_Index, size = RA_Index, alpha = RA_Index)) +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  scale_color_viridis_b() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  labs(title = "eDNA Vertical at Genus Level")

```

# 5. Vertical comparison eDNA index

In this section we explore how well eDNA captures the vertical distribution of zooplankton, compared to the net samples. We use the vertical strudy from Zoopsprint 2022.

First: Rephrase the collapseRAindex function to also handle different sites.:

```{r}
collapseRAindexVert <- function(data) {
  
  output <- data %>%
    
    # Collapse dataset per date, depth and taxa.
    mutate(Abundance = ifelse(is.na(Abundance), 0, Abundance)) %>% 
    group_by(Date, Taxa, Depth, Site_ID) %>% 
    summarise(Abundance = sum(Abundance)) %>% 
    
    # Calculate relative abundance
    group_by(Date, Depth, Site_ID) %>% 
    reframe(RA = Abundance/sum(Abundance),
            Taxa, Abundance) %>%
    filter(is.na(RA) == FALSE) %>%
    
    #Remove outliers before calculating Index:
    group_by(Taxa) %>%
    #filter(RA <= quantile(RA, 0.99)) %>% 
    reframe(RA = ifelse(RA <= quantile(RA, 0.99), RA, -1),
            Date, Depth, Abundance, Site_ID) %>%
    group_by(Taxa) %>%
    reframe(RA = ifelse(RA == -1, max(RA), RA),
            Date, Depth, Abundance, Site_ID) %>% 
    
    #Calculate Index
    group_by(Taxa) %>%
    reframe(Site_ID, Date, Depth, Abundance, RA,
            RA_Index = RA/max(RA)) %>% 
    filter(is.na(RA_Index) == FALSE)

  return(output)
}
```

## Zooplankton net samples:

Microscopic counts:

```{r}
net_microscopy_index <-
  df_count_vertical %>%
  mutate(Taxa = Genus,
         Abundance = Count*Fraction,
         Depth= Depth,
         Site_ID = StationID,
         Date = Sample_Date) %>% 
  # Calculate RA index
  collapseRAindexVert() %>% 
  mutate(Marker = "Microscopy")
```

Make a list of counted taxa:

```{r}
counted_taxa_vertical <-
  net_microscopy_index %>% 
  pull(Taxa) %>% 
  unique()

counted_taxa_vertical
```

18S data:

```{r}

Net_18S_index <-
  df_18S %>% 
  filter(Project_name == "Vertical",
         year(Sample_date) == 2022,
         Division == "Metazoa",
         Sample_type == "Zp_Net",
         is.na(Genus) == FALSE,
         Genus %in% counted_taxa_vertical) %>% 
  mutate(Taxa = Genus,
         Abundance = Abundance,
         Depth= Line_out_depth,
         Date = Sample_date) %>%
  collapseRAindexVert() %>% 
  mutate(Marker = "18S")
```

COI data:

```{r}
Net_COI_index <- 
  df_COI %>%
  filter(Project_name == "Vertical",
         year(Sample_date) == 2022,
         phylum %in% metazoa,
         Sample_type == "Zp_Net",
         is.na(genus) == FALSE,
         genus %in% counted_taxa_vertical) %>%
  mutate(Taxa = genus,
         Abundance = Abundance,
         Depth= Line_out_depth,
         Date = Sample_date) %>% 
  collapseRAindexVert() %>% 
  mutate(Marker = "COI")
```

Combine the three datasets and plot:

```{r}
bind_rows(Net_COI_index, Net_18S_index, net_microscopy_index) %>%
  mutate(Depth = factor(Depth, levels = c(
           "200-120", "120-80", "80-0"))) %>% 
  ggplot() +
  geom_point(aes(RA_Index, Depth, shape = Site_ID, color = Marker)) +
  stat_summary(aes(RA_Index, Depth, color = Marker, group = Marker),
               fun=mean, geom="line") +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  labs(title = "Zooplankton Net Samples - Vertical")

ggsave("Figure_output/Vertical_Net_Genus_Comp.pdf")
```

## eDNA Samples

Same as above but for eDNA samples:

18S:

```{r}
eDNA_18S_index <- df_18S %>% 
  filter(Project_name == "Vertical",
         year(Sample_date) == 2022,
         Division == "Metazoa",
         Sample_type == "eDNA",
         is.na(Genus) == FALSE) %>%
  
  mutate(Taxa = Genus,
         Abundance = Abundance,
         Depth= Line_out_depth,
         Date = Sample_date) %>%
  collapseRAindexVert() %>%
  filter(Taxa %in% counted_taxa_vertical) %>% 
  mutate(Marker = "18S")
```

COI:

```{r}
eDNA_COI_index <- df_COI %>%
  filter(Project_name == "Vertical",
         year(Sample_date) == 2022,
         phylum %in% metazoa,
         Sample_type == "eDNA",
         is.na(genus) == FALSE) %>%
  mutate(Taxa = genus,
         Abundance = Abundance,
         Depth= Line_out_depth,
         Date = Sample_date) %>%
  collapseRAindexVert() %>%
  filter(Taxa %in% counted_taxa_vertical) %>% 
  mutate(Marker = "COI")
```

Combine and plot:

```{r}
bind_rows(eDNA_COI_index, eDNA_18S_index) %>%
  mutate(Depth = factor(Depth, levels = c(
           "160", "120", "100", "80", "40"))) %>% 
  ggplot() +
  geom_point(aes(RA_Index, Depth, shape = Site_ID, color = Marker)) +
  stat_summary(aes(RA_Index, Depth, color = Marker, group = Marker),
               fun=mean, geom="line") +
  facet_wrap(facets = "Taxa") +
  theme_bw() +
  labs(title = "eDNA samples")

ggsave("Figure_output/Vertical_eDNA_Genus_Comp.pdf")

eDNA_COI_index
```

Compare the two plots:

```{r}
comp_plot
eDNA_plot
```
